In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from tqdm import tqdm
import missingno as msno
import seaborn as sns
import scipy.stats as st
from sklearn.ensemble import RandomForestRegressor
from sklearn.model_selection import train_test_split
import xgboost as xg
from lightgbm import LGBMClassifier
from catboost import CatBoostClassifier
from sklearn.metrics import accuracy_score, f1_score
from eli5.sklearn import PermutationImportance
from eli5 import show_weights
from sklearn.tree import DecisionTreeClassifier
from bayes_opt import BayesianOptimization

In [2]:
#Загрузим данные из файла.
df = pd.read_excel('/home/rik/Рабочий стол/МИФИ/учеба/мо/curse/data.xlsx', index_col=0)
df.head()

IC50, mM    CC50, mM         SI  MaxAbsEStateIndex  MaxEStateIndex  \
0    6.239374  175.482382  28.125000           5.094096        5.094096   
1    0.771831    5.402819   7.000000           3.961417        3.961417   
2  223.808778  161.142320   0.720000           2.627117        2.627117   
3    1.705624  107.855654  63.235294           5.097360        5.097360   
4  107.131532  139.270991   1.300000           5.150510        5.150510   

   MinAbsEStateIndex  MinEStateIndex       qed        SPS    MolWt  ...  \
0           0.387225        0.387225  0.417362  42.928571  384.652  ...   
1           0.533868        0.533868  0.462473  45.214286  388.684  ...   
2           0.543231        0.543231  0.260923  42.187500  446.808  ...   
3           0.390603        0.390603  0.377846  41.862069  398.679  ...   
4           0.270476        0.270476  0.429038  36.514286  466.713  ...   

   fr_sulfide  fr_sulfonamd  fr_sulfone  fr_term_acetylene  fr_tetrazole  \
0           0             0           0                  0             0   
1           0             0           0                  0             0   
2           0             0           0                  0             0   
3           0             0           0                  0             0   
4           0             0           0                  0             0   

   fr_thiazole  fr_thiocyan  fr_thiophene  fr_unbrch_alkane  fr_urea  
0            0            0             0                 3        0  
1            0            0             0                 3        0  
2            0            0             0                 3        0  
3            0            0             0                 4        0  
4            0            0             0                 0        0  

[5 rows x 213 columns]

In [3]:
print("train.shape = {} rows, {} cols".format(*df.shape))

train.shape = 1001 rows, 213 cols


# Предобработка данных.

In [4]:
def transform(df): 
    discrete_feature = [feature for feature in df.columns if len(df[feature].unique())<25]
    count_discrete_feature = {}
    for i in tqdm(discrete_feature):
        cnt = 0
        for j in df[i]:
            if j != 0:
                cnt += 1
        count_discrete_feature[f'{i}'] = cnt
    count_discrete_feature = pd.Series(count_discrete_feature)
    df = df.drop(columns=count_discrete_feature[count_discrete_feature == 0].index.to_list())
    df = df.dropna()
    feature_del = ['BertzCT', 'Chi0', 'Chi0v', 'Chi1', 'Chi1n', 'Chi1v', 'Chi2v', 'Chi3n', 'Chi3v', 'Chi4n', 'Chi4v',
     'ExactMolWt', 'FpDensityMorgan1', 'FpDensityMorgan3', 'HeavyAtomCount', 'HeavyAtomMolWt', 'Kappa1', 'Kappa2',
     'Kappa3', 'LabuteASA', 'MaxAbsEStateIndex', 'MaxAbsPartialCharge', 'MolMR', 'MolWt', 'NHOHCount', 'NumAromaticCarbocycles',
     'NumHAcceptors', 'NumHeteroatoms', 'NumSaturatedCarbocycles', 'NumValenceElectrons', 'SMR_VSA4', 'SMR_VSA9',
     'SlogP_VSA6', 'TPSA', 'VSA_EState2', 'VSA_EState3', 'VSA_EState6', 'fr_Al_OH_noTert', 'fr_COO', 'fr_COO2',
     'fr_C_O_noCOO', 'fr_Nhpyrrole', 'fr_benzene', 'fr_nitro_arom_nonortho', 'fr_phenol', 'fr_phenol_noOrthoHbond',
    'MinAbsPartialCharge', 'Chi2n']
    df = df.drop(columns=feature_del)
    df =  df[df['SI'] < 250]
    
    return df

In [5]:
df = transform(df)

100%|███████████████████████████████████████| 110/110 [00:00<00:00, 6459.19it/s]


In [6]:
df['SI'] = df['SI'].apply(lambda x: 0 if x < 8 else 1)
df['SI'].head()

0    1
1    0
2    0
3    1
4    0
Name: SI, dtype: int64

## Обучение нескольких моделей, их сравнение. 

In [7]:
target = df['SI']
data = df.drop(['CC50, mM','IC50, mM','SI'], axis=1)

In [8]:
x_train, x_valid = train_test_split(
    data, train_size=0.8, random_state=1
)
y_train, y_valid = train_test_split(
    target, train_size=0.8, random_state=1
)
print("x_train.shape = {} rows, {} cols".format(*x_train.shape))
print("x_valid.shape = {} rows, {} cols".format(*x_valid.shape))

x_train.shape = 776 rows, 144 cols
x_valid.shape = 195 rows, 144 cols


In [9]:
# Создадим модель дерева решений
tree = DecisionTreeClassifier(random_state=27)
# Создадим модель CatBoost
model = CatBoostClassifier(random_state=27)
# Создадим модель lightgbm
model_LGBM = LGBMClassifier(random_state=27)
# Создадим модель xgboost
xgb_r = xg.XGBClassifier() 
# обучение моделей
tree.fit(x_train, y_train)
model.fit(x_train, y_train, verbose=0)
model_LGBM.fit(x_train, y_train, verbose=0)
xgb_r.fit(x_train, y_train, verbose=0)
# предсказание ответов для тренеровочной выборки
y_pred_tree_train = tree.predict(x_train)
y_pred_сat_Boost_train = model.predict(x_train)
y_pred_LGBM_train = model_LGBM.predict(x_train)
y_pred_xgb_train = xgb_r.predict(x_train)
# предсказание ответов для тестовой выборки
y_pred_tree = tree.predict(x_valid)
y_pred_сat_Boost = model.predict(x_valid)
y_pred_LGBM = model_LGBM.predict(x_valid)
y_pred_xgb = xgb_r.predict(x_valid)

print("Train tree f1_score: ", f1_score(y_train, y_pred_tree_train))
print("Train tree accuracy_score: ", accuracy_score(y_train, y_pred_tree_train))
print("Validation tree f1_score: ", f1_score(y_valid, y_pred_tree))
print("Validation tree accuracy_score: ", accuracy_score(y_valid, y_pred_tree))
print('*' * 50)
print("Train сat_Boost f1_score: ", f1_score(y_train, y_pred_сat_Boost_train))
print("Train сat_Boost accuracy_score: ", accuracy_score(y_train, y_pred_сat_Boost_train))
print("Validation сat_Boost f1_score: ", f1_score(y_valid, y_pred_сat_Boost))
print("Validation сat_Boost accuracy_score: ", accuracy_score(y_valid, y_pred_сat_Boost))
print('*' * 50)
print("Train LGBM f1_score: ", f1_score(y_train, y_pred_LGBM_train))
print("Train LGBM accuracy_score: ", accuracy_score(y_train, y_pred_LGBM_train))
print("Validation LGBM f1_score: ", f1_score(y_valid, y_pred_LGBM))
print("Validation LGBM accuracy_score: ", accuracy_score(y_valid, y_pred_LGBM))
print('*' * 50)
print("Train xgb f1_score: ", f1_score(y_train, y_pred_xgb_train))
print("Train xgb accuracy_score: ", accuracy_score(y_train, y_pred_xgb_train))
print("Validation xgb f1_score: ", f1_score(y_valid, y_pred_xgb))
print("Validation xgb accuracy_score: ", accuracy_score(y_valid, y_pred_xgb))

/usr/local/lib/python3.8/dist-packages/lightgbm/sklearn.py:736: UserWarning: 'verbose' argument is deprecated and will be removed in a future release of LightGBM. Pass 'log_evaluation()' callback via 'callbacks' argument instead.
  _log_warning("'verbose' argument is deprecated and will be removed in a future release of LightGBM. "


Train tree f1_score:  0.9130434782608695
Train tree accuracy_score:  0.9432989690721649
Validation tree f1_score:  0.44827586206896547
Validation tree accuracy_score:  0.6717948717948717
**************************************************
Train сat_Boost f1_score:  0.842323651452282
Train сat_Boost accuracy_score:  0.9020618556701031
Validation сat_Boost f1_score:  0.5321100917431192
Validation сat_Boost accuracy_score:  0.7384615384615385
**************************************************
Train LGBM f1_score:  0.915057915057915
Train LGBM accuracy_score:  0.9432989690721649
Validation LGBM f1_score:  0.5535714285714285
Validation LGBM accuracy_score:  0.7435897435897436
**************************************************
Train xgb f1_score:  0.9153846153846155
Train xgb accuracy_score:  0.9432989690721649
Validation xgb f1_score:  0.543859649122807
Validation xgb accuracy_score:  0.7333333333333333


Из приведенных данных выше мы видим низкое и среднее качество моделей, а также переобучение. Наилучшие результаты показали модели сat_Boost, LGBM, xgb. С ними и продолжим работать.

# Подбор признаков, их анализ и оценка важности.

**CatBoost**

In [10]:
col = x_train.columns.tolist()
estimator_cat_boost = PermutationImportance(
    model, scoring= "f1", random_state=27
)
importance_cat_boost = estimator_cat_boost.fit(x_train, y_train)

In [11]:
show_weights(importance_cat_boost, feature_names=col, top=x_train.shape[1])

Оставим только те признаки которые которые улучшают работу модели. И посмотрим как изменится метрика

In [12]:
feature_cat_boost = [
'MinAbsEStateIndex','VSA_EState4','EState_VSA7','PEOE_VSA6','PEOE_VSA8','FpDensityMorgan2','EState_VSA3','Ipc',
'SMR_VSA5','BCUT2D_MRHI','MolLogP','SlogP_VSA4','VSA_EState8','EState_VSA6','BCUT2D_CHGHI','PEOE_VSA11','MaxEStateIndex',
'Chi0n','qed','VSA_EState10','EState_VSA4','MinPartialCharge','EState_VSA2','fr_aryl_methyl','SlogP_VSA5',
'EState_VSA8','MinEStateIndex','PEOE_VSA12','PEOE_VSA9','PEOE_VSA2','SlogP_VSA2','PEOE_VSA1','EState_VSA9',
'SMR_VSA7','SlogP_VSA10','fr_ester','NumRotatableBonds','MaxPartialCharge','fr_bicyclic','PEOE_VSA14','BCUT2D_LOGPLOW',
'PEOE_VSA7','VSA_EState7','SMR_VSA10','BCUT2D_MWHI','PEOE_VSA3','SlogP_VSA12','AvgIpc','BCUT2D_MRLOW','RingCount',
'PEOE_VSA4','fr_priamide','PEOE_VSA10','fr_ketone','fr_C_O','NumAliphaticHeterocycles','EState_VSA10','NumAliphaticRings',
'SMR_VSA1','SlogP_VSA7','fr_pyridine','SMR_VSA3','HallKierAlpha','fr_aniline','PEOE_VSA5','SlogP_VSA8','fr_para_hydroxylation',
'fr_methoxy','NumAromaticRings','NOCount','NumAliphaticCarbocycles','SPS','fr_unbrch_alkane','VSA_EState9',
'fr_halogen','fr_imidazole','fr_Ar_NH','fr_quatN','EState_VSA5','SlogP_VSA1','fr_thiazole','fr_nitro','fr_allylic_oxid',
'fr_amide']

In [13]:
target = df['SI']
data_cb = data[feature_cat_boost]

In [14]:
x_train_cb, x_valid_cb = train_test_split(
    data_cb, train_size=0.8, random_state=1
)
y_train_cb, y_valid_cb = train_test_split(
    target, train_size=0.8, random_state=1
)
print("x_train.shape = {} rows, {} cols".format(*x_train_cb.shape))
print("x_valid.shape = {} rows, {} cols".format(*x_valid_cb.shape))

x_train.shape = 776 rows, 84 cols
x_valid.shape = 195 rows, 84 cols


In [15]:
# Создадим модель CatBoost
model = CatBoostClassifier(random_state=27)
# обучение моделей
model.fit(x_train_cb, y_train_cb, verbose=0)
# предсказание ответов для тренеровочной выборки
y_pred_сat_Boost_train = model.predict(x_train_cb)
# предсказание ответов для тестовой выборки
y_pred_сat_Boost = model.predict(x_valid_cb)


print("Train сat_Boost f1_score: ", f1_score(y_train_cb, y_pred_сat_Boost_train))
print("Train сat_Boost accuracy_score: ", accuracy_score(y_train_cb, y_pred_сat_Boost_train))
print("Validation сat_Boost f1_score: ", f1_score(y_valid_cb, y_pred_сat_Boost))
print("Validation сat_Boost accuracy_score: ", accuracy_score(y_valid_cb, y_pred_сat_Boost))


Train сat_Boost f1_score:  0.8594704684317719
Train сat_Boost accuracy_score:  0.9110824742268041
Validation сat_Boost f1_score:  0.5504587155963303
Validation сat_Boost accuracy_score:  0.7487179487179487


Отобрав признаки мы немного улучшили метрики для модели CatBoost на вадидации f1_score:  0.55, accuracy_score:  0.75.

**LGBM**

In [16]:
col = x_train.columns.tolist()
estimator_LGBM = PermutationImportance(
    model_LGBM, scoring= "f1", random_state=27
)
importance_LGBM = estimator_LGBM.fit(x_train, y_train)


In [17]:
show_weights(importance_LGBM, feature_names=col, top=x_train.shape[1])

Оставим только те признаки которые которые улучшают работу модели. И посмотрим как изменится метрика

In [18]:
feature_LGBM = ['MaxEStateIndex','MinPartialCharge','MolLogP','FractionCSP3','BCUT2D_CHGLO','VSA_EState4',
'HallKierAlpha','BCUT2D_MRLOW','qed','BCUT2D_LOGPHI','EState_VSA9','MinEStateIndex','NumAromaticRings',
'AvgIpc','SPS','RingCount','PEOE_VSA2','PEOE_VSA5','BCUT2D_MWLOW','EState_VSA2','BCUT2D_MWHI','MinAbsEStateIndex',
'SlogP_VSA1','EState_VSA4','fr_Ar_N','PEOE_VSA11','BCUT2D_MRHI','NumSaturatedRings','EState_VSA3','PEOE_VSA9',
'EState_VSA5','PEOE_VSA3','BalabanJ','PEOE_VSA1','fr_ester','NumRotatableBonds','SMR_VSA1','SMR_VSA10','VSA_EState1',
'VSA_EState10','EState_VSA8','PEOE_VSA12','VSA_EState7','EState_VSA10','BCUT2D_LOGPLOW','PEOE_VSA14','NumHDonors',
'NumAliphaticHeterocycles']

In [19]:
target_LGBM = df['SI']
data_LGBM = data[feature_LGBM]

In [20]:
x_train_LGBM, x_valid_LGBM = train_test_split(
    data_LGBM, train_size=0.8, random_state=1
)
y_train_LGBM, y_valid_LGBM = train_test_split(
    target_LGBM, train_size=0.8, random_state=1
)
print("x_train.shape = {} rows, {} cols".format(*x_train_LGBM.shape))
print("x_valid.shape = {} rows, {} cols".format(*x_valid_LGBM.shape))

x_train.shape = 776 rows, 48 cols
x_valid.shape = 195 rows, 48 cols


In [21]:
model_LGBM = LGBMClassifier(random_state=27)
model_LGBM.fit(x_train_LGBM, y_train_LGBM)
# предсказание ответов для тренеровочной выборки
y_pred_LGBM_train = model_LGBM.predict(x_train_LGBM)
# предсказание ответов для тестовой выборки
y_pred_LGBM = model_LGBM.predict(x_valid_LGBM)

print("Train LGBM f1_score: ", f1_score(y_train_LGBM, y_pred_LGBM_train))
print("Train LGBM accuracy_score: ", accuracy_score(y_train_LGBM, y_pred_LGBM_train))
print("Validation LGBM f1_score: ", f1_score(y_valid_LGBM, y_pred_LGBM))
print("Validation LGBM accuracy_score: ", accuracy_score(y_valid_LGBM, y_pred_LGBM))

Train LGBM f1_score:  0.914396887159533
Train LGBM accuracy_score:  0.9432989690721649
Validation LGBM f1_score:  0.5688073394495413
Validation LGBM accuracy_score:  0.7589743589743589


Отобрав признаки мы улучшили метрики для модели LGBMClassifier на вадидации f1_score:  0.57, accuracy_score:  0.76.

**xgboost**

In [22]:
col = x_train.columns.tolist()
estimator_xgb = PermutationImportance(
    xgb_r, scoring= "neg_mean_squared_error", random_state=27
)
importance_xgb = estimator_xgb.fit(x_train, y_train)

In [23]:
show_weights(importance_xgb, feature_names=col, top=x_train.shape[1])

Оставим только те признаки которые которые улучшают работу модели. И посмотрим как изменится метрика

In [24]:
feature_xgb = ['BCUT2D_MRLOW','BCUT2D_LOGPHI','MinPartialCharge','MolLogP','MinAbsEStateIndex','VSA_EState4',
'PEOE_VSA9','PEOE_VSA8','EState_VSA2','MinEStateIndex','AvgIpc','EState_VSA4','PEOE_VSA3','VSA_EState8',
'MaxEStateIndex','FractionCSP3','BCUT2D_MRHI','SMR_VSA10','PEOE_VSA6','qed','Chi0n','EState_VSA7','SPS','PEOE_VSA10',
'RingCount','EState_VSA3','BCUT2D_MWLOW','HallKierAlpha','fr_C_O','fr_NH0','SlogP_VSA1','VSA_EState10','BCUT2D_CHGLO',
'NOCount','SlogP_VSA4','fr_NH1','BCUT2D_CHGHI','VSA_EState9','PEOE_VSA5','BCUT2D_MWHI','NumSaturatedHeterocycles',
'NumAliphaticCarbocycles','EState_VSA9','Ipc','VSA_EState1','EState_VSA10','VSA_EState7','SMR_VSA1','NumRotatableBonds',
'BCUT2D_LOGPLOW','SlogP_VSA3','FpDensityMorgan2']

In [25]:
target_xgb = df['SI']
data_xgb = data[feature_xgb]

In [26]:
x_train_xgb, x_valid_xgb = train_test_split(
    data_xgb, train_size=0.8, random_state=1
)
y_train_xgb, y_valid_xgb = train_test_split(
    target_xgb, train_size=0.8, random_state=1
)
print("x_train.shape = {} rows, {} cols".format(*x_train_xgb.shape))
print("x_valid.shape = {} rows, {} cols".format(*x_valid_xgb.shape))

x_train.shape = 776 rows, 52 cols
x_valid.shape = 195 rows, 52 cols


In [27]:
model_xgb = xg.XGBClassifier(random_state=27)
model_xgb.fit(x_train_xgb, y_train_xgb, verbose=100)

XGBClassifier(base_score=None, booster=None, callbacks=None,
              colsample_bylevel=None, colsample_bynode=None,
              colsample_bytree=None, device=None, early_stopping_rounds=None,
              enable_categorical=False, eval_metric=None, feature_types=None,
              gamma=None, grow_policy=None, importance_type=None,
              interaction_constraints=None, learning_rate=None, max_bin=None,
              max_cat_threshold=None, max_cat_to_onehot=None,
              max_delta_step=None, max_depth=None, max_leaves=None,
              min_child_weight=None, missing=nan, monotone_constraints=None,
              multi_strategy=None, n_estimators=None, n_jobs=None,
              num_parallel_tree=None, random_state=27, ...)

In [28]:
y_pred_xgb = model_xgb.predict(x_train_xgb)
y_val_xgb = model_xgb.predict(x_valid_xgb)
print("Train f1_score: ", f1_score(y_train_xgb, y_pred_xgb))
print("Train accuracy_score: ", accuracy_score(y_train_xgb, y_pred_xgb))
print("Validation f1_score: ", f1_score(y_valid_xgb, y_val_xgb))
print("Validation accuracy_score: ", accuracy_score(y_valid_xgb, y_val_xgb))

Train f1_score:  0.9140625
Train accuracy_score:  0.9432989690721649
Validation f1_score:  0.5913043478260869
Validation accuracy_score:  0.7589743589743589


Отобрав признаки мы улучшили метрики для модели xgboost на вадидации f1_score:  0.59, accuracy_score:  0.76.

# Подбор гиперпараметров

**CatBoost**

In [29]:
def catboost_cv(depth, learning_rate,n_estimators):
    depth = int(depth)
    n_estimators = int(n_estimators)
    model = CatBoostClassifier(
        depth=depth,
        learning_rate=learning_rate,
        n_estimators=n_estimators,
        eval_metric  = 'F1',
        random_seed=27
    )
    model.fit(x_train_cb, y_train_cb, verbose=100)

    return f1_score(y_valid_cb, model.predict(x_valid_cb))

In [30]:
param_space = {
    'depth': (2, 10),             
    'learning_rate': (0.001, 0.3),
    'n_estimators': (50, 1000)
}

bayesian_opt = BayesianOptimization(
    f=catboost_cv, pbounds=param_space, random_state=27)
bayesian_opt.maximize(init_points=5, n_iter=20)
results = pd.DataFrame(bayesian_opt.res)
results.sort_values(by='target', ascending=False, inplace=True)

|   iter    |  target   |   depth   | learni... | n_esti... |
-------------------------------------------------------------
0:	learn: 0.3860590	total: 19.6ms	remaining: 14.6s
100:	learn: 0.8986083	total: 491ms	remaining: 3.15s
200:	learn: 0.9147287	total: 899ms	remaining: 2.45s
300:	learn: 0.9147287	total: 1.31s	remaining: 1.94s
400:	learn: 0.9160305	total: 1.78s	remaining: 1.54s
500:	learn: 0.9157088	total: 2.25s	remaining: 1.11s
600:	learn: 0.9147287	total: 2.78s	remaining: 681ms
700:	learn: 0.9147287	total: 3.2s	remaining: 215ms
747:	learn: 0.9163498	total: 3.39s	remaining: 0us
| 1         | 0.5263    | 5.406     | 0.2446    | 748.6     |
0:	learn: 0.5058548	total: 25.4ms	remaining: 24.8s
100:	learn: 0.9139579	total: 2.54s	remaining: 22.2s
200:	learn: 0.9147287	total: 5.01s	remaining: 19.4s
300:	learn: 0.9147287	total: 7.33s	remaining: 16.5s
400:	learn: 0.9150579	total: 9.66s	remaining: 13.9s
500:	learn: 0.9143969	total: 12.1s	remaining: 11.5s
600:	learn: 0.9147287	total: 14.4s	rema

| 21        | 0.5424    | 2.697     | 0.1964    | 340.2     |
0:	learn: 0.3860590	total: 7.64ms	remaining: 6.61s
100:	learn: 0.9129594	total: 655ms	remaining: 4.96s
200:	learn: 0.9160305	total: 1.3s	remaining: 4.3s
300:	learn: 0.9157088	total: 1.95s	remaining: 3.65s
400:	learn: 0.9153846	total: 2.6s	remaining: 3.01s
500:	learn: 0.9160305	total: 3.25s	remaining: 2.36s
600:	learn: 0.9150579	total: 3.95s	remaining: 1.73s
700:	learn: 0.9143969	total: 4.6s	remaining: 1.07s
800:	learn: 0.9147287	total: 5.26s	remaining: 420ms
864:	learn: 0.9160305	total: 5.7s	remaining: 0us
| 22        | 0.5133    | 6.781     | 0.172     | 865.4     |
0:	learn: 0.4263158	total: 12.9ms	remaining: 2.91s
100:	learn: 0.9153846	total: 1.17s	remaining: 1.45s
200:	learn: 0.9140625	total: 2.33s	remaining: 290ms
225:	learn: 0.9153846	total: 2.63s	remaining: 0us
| 23        | 0.4956    | 7.484     | 0.2354    | 226.6     |
0:	learn: 0.0000000	total: 3.06ms	remaining: 1.03s
100:	learn: 0.2687500	total: 141ms	remaining: 

In [31]:
best_hyperparameters = bayesian_opt.max
best_hyperparameters['params'] = {param: int(value) if param in [
    'depth', 'iterations', 'l2_leaf_reg'] else value for param, value in best_hyperparameters['params'].items()}
print("Best hyperparameters:", best_hyperparameters['params'])
print(f"Best R-squared Score: {best_hyperparameters['target']:.4f}")

Best hyperparameters: {'depth': 9, 'learning_rate': 0.2005611810761974, 'n_estimators': 227.93938171457134}
Best R-squared Score: 0.5841


In [32]:
params = {
    'depth': 9,             
    'learning_rate': 0.2005611810761974, 
    'n_estimators': 228
}
model = CatBoostClassifier(**params, random_state=27)
model.fit(x_train_cb, y_train_cb, verbose=100)
y_pred_cb = model.predict(x_train_cb)
y_val_cb = model.predict(x_valid_cb)
print("Train сat_Boost f1_score: ", f1_score(y_train_cb, y_pred_cb))
print("Train сat_Boost accuracy_score: ", accuracy_score(y_train_cb, y_pred_cb))
print("Validation сat_Boost f1_score: ", f1_score(y_valid_cb, y_val_cb))
print("Validation сat_Boost accuracy_score: ", accuracy_score(y_valid_cb, y_val_cb))

0:	learn: 0.6237191	total: 53.6ms	remaining: 12.2s
100:	learn: 0.1201141	total: 4.76s	remaining: 5.98s
200:	learn: 0.1012288	total: 9.48s	remaining: 1.27s
227:	learn: 0.0995476	total: 10.8s	remaining: 0us
Train сat_Boost f1_score:  0.9153846153846155
Train сat_Boost accuracy_score:  0.9432989690721649
Validation сat_Boost f1_score:  0.584070796460177
Validation сat_Boost accuracy_score:  0.7589743589743589


После подбора гиперпараметров мы улучшили метрики модели CatBoost на вадидации f1_score:  0.58, accuracy_score:  0.76.

**LGBM**

In [33]:
def LGBM_cv(max_depth, learning_rate, n_estimators):
    max_depth = int(max_depth)
    n_estimators = int(n_estimators)
    model = LGBMClassifier(
        max_depth=max_depth,
        learning_rate=learning_rate,
        n_estimators=n_estimators,          
        eval_metric  = 'F1',
        random_seed=27
    )
    model.fit(x_train_LGBM, y_train_LGBM, eval_set = (x_valid_LGBM, y_valid_LGBM))
    return f1_score(y_valid_LGBM, model.predict(x_valid_LGBM))

In [34]:
param_space = {
    'max_depth': (2, 10),            
    'learning_rate': (0.001, 0.3),
    'n_estimators': (50, 1000)
}

bayesian_opt = BayesianOptimization(
    f=LGBM_cv, pbounds=param_space, random_state=27)
bayesian_opt.maximize(init_points=5, n_iter=25)
results = pd.DataFrame(bayesian_opt.res)
results.sort_values(by='target', ascending=False, inplace=True)

|   iter    |  target   | learni... | max_depth | n_esti... |
-------------------------------------------------------------
[LightGBM] [Warning] Unknown parameter: eval_metric
[1]	valid_0's binary_logloss: 0.608123
[2]	valid_0's binary_logloss: 0.598006
[3]	valid_0's binary_logloss: 0.588786
[4]	valid_0's binary_logloss: 0.58126
[5]	valid_0's binary_logloss: 0.576983
[6]	valid_0's binary_logloss: 0.573339
[7]	valid_0's binary_logloss: 0.56972
[8]	valid_0's binary_logloss: 0.570631
[9]	valid_0's binary_logloss: 0.57009
[10]	valid_0's binary_logloss: 0.575064
[11]	valid_0's binary_logloss: 0.573994
[12]	valid_0's binary_logloss: 0.572611
[13]	valid_0's binary_logloss: 0.57134
[14]	valid_0's binary_logloss: 0.573504
[15]	valid_0's binary_logloss: 0.578493
[16]	valid_0's binary_logloss: 0.583252
[17]	valid_0's binary_logloss: 0.58023
[18]	valid_0's binary_logloss: 0.582279
[19]	valid_0's binary_logloss: 0.579354
[20]	valid_0's binary_logloss: 0.577693
[21]	valid_0's binary_logloss: 0.57703

[274]	valid_0's binary_logloss: 1.15007
[275]	valid_0's binary_logloss: 1.15086
[276]	valid_0's binary_logloss: 1.15199
[277]	valid_0's binary_logloss: 1.15264
[278]	valid_0's binary_logloss: 1.15261
[279]	valid_0's binary_logloss: 1.15304
[280]	valid_0's binary_logloss: 1.15414
[281]	valid_0's binary_logloss: 1.15936
[282]	valid_0's binary_logloss: 1.16175
[283]	valid_0's binary_logloss: 1.16511
[284]	valid_0's binary_logloss: 1.16826
[285]	valid_0's binary_logloss: 1.16807
[286]	valid_0's binary_logloss: 1.16913
[287]	valid_0's binary_logloss: 1.1708
[288]	valid_0's binary_logloss: 1.17245
[289]	valid_0's binary_logloss: 1.17373
[290]	valid_0's binary_logloss: 1.17431
[291]	valid_0's binary_logloss: 1.17692
[292]	valid_0's binary_logloss: 1.17882
[293]	valid_0's binary_logloss: 1.17841
[294]	valid_0's binary_logloss: 1.17924
[295]	valid_0's binary_logloss: 1.18111
[296]	valid_0's binary_logloss: 1.18378
[297]	valid_0's binary_logloss: 1.18762
[298]	valid_0's binary_logloss: 1.18738
[

[574]	valid_0's binary_logloss: 1.44456
[575]	valid_0's binary_logloss: 1.44511
[576]	valid_0's binary_logloss: 1.44616
[577]	valid_0's binary_logloss: 1.44631
[578]	valid_0's binary_logloss: 1.4451
[579]	valid_0's binary_logloss: 1.44532
[580]	valid_0's binary_logloss: 1.44522
[581]	valid_0's binary_logloss: 1.44673
[582]	valid_0's binary_logloss: 1.44702
[583]	valid_0's binary_logloss: 1.44839
[584]	valid_0's binary_logloss: 1.44878
[585]	valid_0's binary_logloss: 1.44925
[586]	valid_0's binary_logloss: 1.45014
[587]	valid_0's binary_logloss: 1.45072
[588]	valid_0's binary_logloss: 1.45145
[589]	valid_0's binary_logloss: 1.45147
[590]	valid_0's binary_logloss: 1.45264
[591]	valid_0's binary_logloss: 1.45163
[592]	valid_0's binary_logloss: 1.45152
[593]	valid_0's binary_logloss: 1.45386
[594]	valid_0's binary_logloss: 1.45452
[595]	valid_0's binary_logloss: 1.45651
[596]	valid_0's binary_logloss: 1.45681
[597]	valid_0's binary_logloss: 1.45777
[598]	valid_0's binary_logloss: 1.45893
[

[122]	valid_0's binary_logloss: 0.926972
[123]	valid_0's binary_logloss: 0.932354
[124]	valid_0's binary_logloss: 0.935301
[125]	valid_0's binary_logloss: 0.940111
[126]	valid_0's binary_logloss: 0.943541
[127]	valid_0's binary_logloss: 0.944808
[128]	valid_0's binary_logloss: 0.945579
[129]	valid_0's binary_logloss: 0.9504
[130]	valid_0's binary_logloss: 0.953028
[131]	valid_0's binary_logloss: 0.955523
[132]	valid_0's binary_logloss: 0.959538
[133]	valid_0's binary_logloss: 0.961745
[134]	valid_0's binary_logloss: 0.967126
[135]	valid_0's binary_logloss: 0.969749
[136]	valid_0's binary_logloss: 0.974426
[137]	valid_0's binary_logloss: 0.975812
[138]	valid_0's binary_logloss: 0.97839
[139]	valid_0's binary_logloss: 0.984873
[140]	valid_0's binary_logloss: 0.983015
[141]	valid_0's binary_logloss: 0.987047
[142]	valid_0's binary_logloss: 0.988474
[143]	valid_0's binary_logloss: 0.992185
[144]	valid_0's binary_logloss: 0.994093
[145]	valid_0's binary_logloss: 0.997141
[146]	valid_0's bin

[370]	valid_0's binary_logloss: 1.42986
[371]	valid_0's binary_logloss: 1.43201
[372]	valid_0's binary_logloss: 1.43478
[373]	valid_0's binary_logloss: 1.43615
[374]	valid_0's binary_logloss: 1.44173
[375]	valid_0's binary_logloss: 1.44439
[376]	valid_0's binary_logloss: 1.44223
[377]	valid_0's binary_logloss: 1.44237
[378]	valid_0's binary_logloss: 1.44678
[379]	valid_0's binary_logloss: 1.44944
[380]	valid_0's binary_logloss: 1.44991
[381]	valid_0's binary_logloss: 1.44939
[382]	valid_0's binary_logloss: 1.45172
[383]	valid_0's binary_logloss: 1.45172
[384]	valid_0's binary_logloss: 1.45205
[385]	valid_0's binary_logloss: 1.45173
[386]	valid_0's binary_logloss: 1.45417
[387]	valid_0's binary_logloss: 1.45325
[388]	valid_0's binary_logloss: 1.45573
[389]	valid_0's binary_logloss: 1.4574
[390]	valid_0's binary_logloss: 1.4575
[391]	valid_0's binary_logloss: 1.45735
[392]	valid_0's binary_logloss: 1.45774
[393]	valid_0's binary_logloss: 1.45782
[394]	valid_0's binary_logloss: 1.45882
[3

[595]	valid_0's binary_logloss: 1.58017
[596]	valid_0's binary_logloss: 1.58034
[597]	valid_0's binary_logloss: 1.58061
[598]	valid_0's binary_logloss: 1.58093
[599]	valid_0's binary_logloss: 1.58131
[600]	valid_0's binary_logloss: 1.58307
[601]	valid_0's binary_logloss: 1.58302
[602]	valid_0's binary_logloss: 1.58322
[603]	valid_0's binary_logloss: 1.58299
[604]	valid_0's binary_logloss: 1.58303
[605]	valid_0's binary_logloss: 1.58307
[606]	valid_0's binary_logloss: 1.58331
[607]	valid_0's binary_logloss: 1.58347
[608]	valid_0's binary_logloss: 1.58469
[609]	valid_0's binary_logloss: 1.58555
[610]	valid_0's binary_logloss: 1.58609
[611]	valid_0's binary_logloss: 1.58605
[612]	valid_0's binary_logloss: 1.586
[613]	valid_0's binary_logloss: 1.58595
[614]	valid_0's binary_logloss: 1.58623
[615]	valid_0's binary_logloss: 1.58634
[616]	valid_0's binary_logloss: 1.58639
[617]	valid_0's binary_logloss: 1.5865
[618]	valid_0's binary_logloss: 1.58551
[619]	valid_0's binary_logloss: 1.58595
[62

[817]	valid_0's binary_logloss: 1.63624
[818]	valid_0's binary_logloss: 1.63633
[819]	valid_0's binary_logloss: 1.63633
[820]	valid_0's binary_logloss: 1.63643
[821]	valid_0's binary_logloss: 1.63649
[822]	valid_0's binary_logloss: 1.63689
[823]	valid_0's binary_logloss: 1.63698
[824]	valid_0's binary_logloss: 1.63702
[825]	valid_0's binary_logloss: 1.63715
[826]	valid_0's binary_logloss: 1.63725
[827]	valid_0's binary_logloss: 1.6374
[828]	valid_0's binary_logloss: 1.63755
[829]	valid_0's binary_logloss: 1.63769
[830]	valid_0's binary_logloss: 1.63786
[831]	valid_0's binary_logloss: 1.63784
[832]	valid_0's binary_logloss: 1.63778
[833]	valid_0's binary_logloss: 1.63803
[834]	valid_0's binary_logloss: 1.63837
[835]	valid_0's binary_logloss: 1.63863
[836]	valid_0's binary_logloss: 1.6387
[837]	valid_0's binary_logloss: 1.63888
[838]	valid_0's binary_logloss: 1.63882
[839]	valid_0's binary_logloss: 1.63877
[840]	valid_0's binary_logloss: 1.63887
[841]	valid_0's binary_logloss: 1.63911
[8

[361]	valid_0's binary_logloss: 1.04601
[362]	valid_0's binary_logloss: 1.04622
[363]	valid_0's binary_logloss: 1.04618
[364]	valid_0's binary_logloss: 1.04803
[365]	valid_0's binary_logloss: 1.04809
[366]	valid_0's binary_logloss: 1.04888
[367]	valid_0's binary_logloss: 1.05202
[368]	valid_0's binary_logloss: 1.05462
[369]	valid_0's binary_logloss: 1.05635
[370]	valid_0's binary_logloss: 1.05702
[371]	valid_0's binary_logloss: 1.05793
[372]	valid_0's binary_logloss: 1.05965
[373]	valid_0's binary_logloss: 1.05909
[374]	valid_0's binary_logloss: 1.05964
[375]	valid_0's binary_logloss: 1.06354
[376]	valid_0's binary_logloss: 1.06742
[377]	valid_0's binary_logloss: 1.06822
[378]	valid_0's binary_logloss: 1.06669
[379]	valid_0's binary_logloss: 1.07087
[380]	valid_0's binary_logloss: 1.0698
[381]	valid_0's binary_logloss: 1.07014
[382]	valid_0's binary_logloss: 1.07343
[383]	valid_0's binary_logloss: 1.07651
[384]	valid_0's binary_logloss: 1.07802
[385]	valid_0's binary_logloss: 1.07751
[

[704]	valid_0's binary_logloss: 1.37124
[705]	valid_0's binary_logloss: 1.37182
[706]	valid_0's binary_logloss: 1.37108
[707]	valid_0's binary_logloss: 1.37228
[708]	valid_0's binary_logloss: 1.3718
[709]	valid_0's binary_logloss: 1.37265
[710]	valid_0's binary_logloss: 1.37389
[711]	valid_0's binary_logloss: 1.37412
[712]	valid_0's binary_logloss: 1.37408
[713]	valid_0's binary_logloss: 1.37729
[714]	valid_0's binary_logloss: 1.37593
[715]	valid_0's binary_logloss: 1.37632
[716]	valid_0's binary_logloss: 1.37694
[717]	valid_0's binary_logloss: 1.37905
[718]	valid_0's binary_logloss: 1.37854
[719]	valid_0's binary_logloss: 1.37954
[720]	valid_0's binary_logloss: 1.37987
[721]	valid_0's binary_logloss: 1.38033
[722]	valid_0's binary_logloss: 1.38009
[723]	valid_0's binary_logloss: 1.38023
[724]	valid_0's binary_logloss: 1.3805
[725]	valid_0's binary_logloss: 1.38083
[726]	valid_0's binary_logloss: 1.38233
[727]	valid_0's binary_logloss: 1.38301
[728]	valid_0's binary_logloss: 1.38509
[7

[202]	valid_0's binary_logloss: 1.23298
[203]	valid_0's binary_logloss: 1.23488
[204]	valid_0's binary_logloss: 1.24102
[205]	valid_0's binary_logloss: 1.24415
[206]	valid_0's binary_logloss: 1.2452
[207]	valid_0's binary_logloss: 1.24602
[208]	valid_0's binary_logloss: 1.24775
[209]	valid_0's binary_logloss: 1.24897
[210]	valid_0's binary_logloss: 1.2512
[211]	valid_0's binary_logloss: 1.24975
[212]	valid_0's binary_logloss: 1.25156
[213]	valid_0's binary_logloss: 1.25278
[214]	valid_0's binary_logloss: 1.2533
[215]	valid_0's binary_logloss: 1.25294
[216]	valid_0's binary_logloss: 1.25597
[217]	valid_0's binary_logloss: 1.25864
[218]	valid_0's binary_logloss: 1.25976
[219]	valid_0's binary_logloss: 1.26376
[220]	valid_0's binary_logloss: 1.26634
[221]	valid_0's binary_logloss: 1.2672
[222]	valid_0's binary_logloss: 1.27048
[223]	valid_0's binary_logloss: 1.27244
[224]	valid_0's binary_logloss: 1.27805
[225]	valid_0's binary_logloss: 1.27914
[226]	valid_0's binary_logloss: 1.27996
[227

[485]	valid_0's binary_logloss: 1.51847
[486]	valid_0's binary_logloss: 1.51885
[487]	valid_0's binary_logloss: 1.52136
[488]	valid_0's binary_logloss: 1.52187
[489]	valid_0's binary_logloss: 1.52283
[490]	valid_0's binary_logloss: 1.52322
[491]	valid_0's binary_logloss: 1.52327
[492]	valid_0's binary_logloss: 1.52352
[493]	valid_0's binary_logloss: 1.52358
[494]	valid_0's binary_logloss: 1.52394
[495]	valid_0's binary_logloss: 1.52418
[496]	valid_0's binary_logloss: 1.524
[497]	valid_0's binary_logloss: 1.52434
[498]	valid_0's binary_logloss: 1.52473
[499]	valid_0's binary_logloss: 1.52487
[500]	valid_0's binary_logloss: 1.52488
[501]	valid_0's binary_logloss: 1.52875
[502]	valid_0's binary_logloss: 1.52911
[503]	valid_0's binary_logloss: 1.52925
[504]	valid_0's binary_logloss: 1.52942
[505]	valid_0's binary_logloss: 1.52966
[506]	valid_0's binary_logloss: 1.52988
[507]	valid_0's binary_logloss: 1.5303
[508]	valid_0's binary_logloss: 1.53049
[509]	valid_0's binary_logloss: 1.5306
[510

[785]	valid_0's binary_logloss: 1.59579
[786]	valid_0's binary_logloss: 1.59585
[787]	valid_0's binary_logloss: 1.59584
[788]	valid_0's binary_logloss: 1.59584
[789]	valid_0's binary_logloss: 1.59589
[790]	valid_0's binary_logloss: 1.5958
[791]	valid_0's binary_logloss: 1.59591
[792]	valid_0's binary_logloss: 1.59609
[793]	valid_0's binary_logloss: 1.59635
[794]	valid_0's binary_logloss: 1.59636
[795]	valid_0's binary_logloss: 1.59663
[796]	valid_0's binary_logloss: 1.59671
[797]	valid_0's binary_logloss: 1.59679
[798]	valid_0's binary_logloss: 1.59671
[799]	valid_0's binary_logloss: 1.59673
[800]	valid_0's binary_logloss: 1.59695
[801]	valid_0's binary_logloss: 1.59717
[802]	valid_0's binary_logloss: 1.59738
[803]	valid_0's binary_logloss: 1.59768
[804]	valid_0's binary_logloss: 1.59758
[805]	valid_0's binary_logloss: 1.59788
[806]	valid_0's binary_logloss: 1.59788
[807]	valid_0's binary_logloss: 1.59809
[808]	valid_0's binary_logloss: 1.59814
[809]	valid_0's binary_logloss: 1.59794
[

[153]	valid_0's binary_logloss: 1.22795
[154]	valid_0's binary_logloss: 1.23223
[155]	valid_0's binary_logloss: 1.23422
[156]	valid_0's binary_logloss: 1.23331
[157]	valid_0's binary_logloss: 1.23478
[158]	valid_0's binary_logloss: 1.23485
[159]	valid_0's binary_logloss: 1.23871
[160]	valid_0's binary_logloss: 1.24044
[161]	valid_0's binary_logloss: 1.242
[162]	valid_0's binary_logloss: 1.24587
[163]	valid_0's binary_logloss: 1.24871
[164]	valid_0's binary_logloss: 1.2479
[165]	valid_0's binary_logloss: 1.25248
[166]	valid_0's binary_logloss: 1.25703
[167]	valid_0's binary_logloss: 1.25826
[168]	valid_0's binary_logloss: 1.26053
[169]	valid_0's binary_logloss: 1.26371
[170]	valid_0's binary_logloss: 1.26258
[171]	valid_0's binary_logloss: 1.26775
[172]	valid_0's binary_logloss: 1.26991
[173]	valid_0's binary_logloss: 1.2739
[174]	valid_0's binary_logloss: 1.28044
[175]	valid_0's binary_logloss: 1.28036
[176]	valid_0's binary_logloss: 1.28236
[177]	valid_0's binary_logloss: 1.28853
[178

[178]	valid_0's binary_logloss: 1.29138
[179]	valid_0's binary_logloss: 1.29986
[180]	valid_0's binary_logloss: 1.30396
[181]	valid_0's binary_logloss: 1.30346
[182]	valid_0's binary_logloss: 1.30429
[183]	valid_0's binary_logloss: 1.30531
[184]	valid_0's binary_logloss: 1.30606
[185]	valid_0's binary_logloss: 1.30943
[186]	valid_0's binary_logloss: 1.31073
[187]	valid_0's binary_logloss: 1.31451
[188]	valid_0's binary_logloss: 1.31176
[189]	valid_0's binary_logloss: 1.31387
[190]	valid_0's binary_logloss: 1.31547
[191]	valid_0's binary_logloss: 1.31572
[192]	valid_0's binary_logloss: 1.31708
[193]	valid_0's binary_logloss: 1.32004
[194]	valid_0's binary_logloss: 1.32171
[195]	valid_0's binary_logloss: 1.32621
[196]	valid_0's binary_logloss: 1.32228
[197]	valid_0's binary_logloss: 1.33181
[198]	valid_0's binary_logloss: 1.33461
[199]	valid_0's binary_logloss: 1.33772
[200]	valid_0's binary_logloss: 1.3398
[201]	valid_0's binary_logloss: 1.34198
[202]	valid_0's binary_logloss: 1.3427
[2

[522]	valid_0's binary_logloss: 1.61075
[523]	valid_0's binary_logloss: 1.61153
[524]	valid_0's binary_logloss: 1.61193
[525]	valid_0's binary_logloss: 1.61233
[526]	valid_0's binary_logloss: 1.61263
[527]	valid_0's binary_logloss: 1.61291
[528]	valid_0's binary_logloss: 1.61331
[529]	valid_0's binary_logloss: 1.6136
[530]	valid_0's binary_logloss: 1.61385
[531]	valid_0's binary_logloss: 1.61416
[532]	valid_0's binary_logloss: 1.61453
[533]	valid_0's binary_logloss: 1.61454
[534]	valid_0's binary_logloss: 1.61445
[535]	valid_0's binary_logloss: 1.61464
[536]	valid_0's binary_logloss: 1.6149
[537]	valid_0's binary_logloss: 1.61499
[538]	valid_0's binary_logloss: 1.61519
[539]	valid_0's binary_logloss: 1.61782
[540]	valid_0's binary_logloss: 1.61816
[541]	valid_0's binary_logloss: 1.61771
[542]	valid_0's binary_logloss: 1.61833
[543]	valid_0's binary_logloss: 1.6186
[544]	valid_0's binary_logloss: 1.61876
[545]	valid_0's binary_logloss: 1.61906
[546]	valid_0's binary_logloss: 1.61907
[54

[LightGBM] [Warning] Unknown parameter: eval_metric
[1]	valid_0's binary_logloss: 0.615333
[2]	valid_0's binary_logloss: 0.605645
[3]	valid_0's binary_logloss: 0.59903
[4]	valid_0's binary_logloss: 0.592136
[5]	valid_0's binary_logloss: 0.590464
[6]	valid_0's binary_logloss: 0.589101
[7]	valid_0's binary_logloss: 0.584782
[8]	valid_0's binary_logloss: 0.583923
[9]	valid_0's binary_logloss: 0.582487
[10]	valid_0's binary_logloss: 0.580126
[11]	valid_0's binary_logloss: 0.579084
[12]	valid_0's binary_logloss: 0.579483
[13]	valid_0's binary_logloss: 0.581846
[14]	valid_0's binary_logloss: 0.583574
[15]	valid_0's binary_logloss: 0.585006
[16]	valid_0's binary_logloss: 0.58356
[17]	valid_0's binary_logloss: 0.585454
[18]	valid_0's binary_logloss: 0.583961
[19]	valid_0's binary_logloss: 0.584683
[20]	valid_0's binary_logloss: 0.58022
[21]	valid_0's binary_logloss: 0.575208
[22]	valid_0's binary_logloss: 0.572958
[23]	valid_0's binary_logloss: 0.57425
[24]	valid_0's binary_logloss: 0.573378
[

[430]	valid_0's binary_logloss: 1.03504
[431]	valid_0's binary_logloss: 1.03679
[432]	valid_0's binary_logloss: 1.03923
[433]	valid_0's binary_logloss: 1.03955
[434]	valid_0's binary_logloss: 1.03996
[435]	valid_0's binary_logloss: 1.04336
[436]	valid_0's binary_logloss: 1.04362
[437]	valid_0's binary_logloss: 1.04465
[438]	valid_0's binary_logloss: 1.04521
[439]	valid_0's binary_logloss: 1.04636
[440]	valid_0's binary_logloss: 1.04641
[441]	valid_0's binary_logloss: 1.04734
[442]	valid_0's binary_logloss: 1.04836
[443]	valid_0's binary_logloss: 1.0485
[444]	valid_0's binary_logloss: 1.05211
[445]	valid_0's binary_logloss: 1.0531
[446]	valid_0's binary_logloss: 1.0532
[447]	valid_0's binary_logloss: 1.05445
[448]	valid_0's binary_logloss: 1.05369
[449]	valid_0's binary_logloss: 1.0585
[450]	valid_0's binary_logloss: 1.0586
[451]	valid_0's binary_logloss: 1.05913
[452]	valid_0's binary_logloss: 1.05965
[453]	valid_0's binary_logloss: 1.06032
[454]	valid_0's binary_logloss: 1.06286
[455]

[684]	valid_0's binary_logloss: 1.24504
[685]	valid_0's binary_logloss: 1.24485
[686]	valid_0's binary_logloss: 1.24568
[687]	valid_0's binary_logloss: 1.24684
[688]	valid_0's binary_logloss: 1.24685
[689]	valid_0's binary_logloss: 1.24701
[690]	valid_0's binary_logloss: 1.24735
[691]	valid_0's binary_logloss: 1.24765
[692]	valid_0's binary_logloss: 1.24869
[693]	valid_0's binary_logloss: 1.24947
[694]	valid_0's binary_logloss: 1.25105
[695]	valid_0's binary_logloss: 1.25132
[696]	valid_0's binary_logloss: 1.25119
[697]	valid_0's binary_logloss: 1.25131
[698]	valid_0's binary_logloss: 1.25271
[699]	valid_0's binary_logloss: 1.25326
[700]	valid_0's binary_logloss: 1.25485
[701]	valid_0's binary_logloss: 1.25668
[702]	valid_0's binary_logloss: 1.25492
[703]	valid_0's binary_logloss: 1.25713
[704]	valid_0's binary_logloss: 1.25839
[705]	valid_0's binary_logloss: 1.25918
[706]	valid_0's binary_logloss: 1.25954
[707]	valid_0's binary_logloss: 1.26107
[708]	valid_0's binary_logloss: 1.2617
[

| 7         | 0.4828    | 0.09637   | 5.178     | 980.6     |
[LightGBM] [Warning] Unknown parameter: eval_metric
[1]	valid_0's binary_logloss: 0.588902
[2]	valid_0's binary_logloss: 0.581264
[3]	valid_0's binary_logloss: 0.575871
[4]	valid_0's binary_logloss: 0.580266
[5]	valid_0's binary_logloss: 0.579293
[6]	valid_0's binary_logloss: 0.584266
[7]	valid_0's binary_logloss: 0.596566
[8]	valid_0's binary_logloss: 0.604139
[9]	valid_0's binary_logloss: 0.59889
[10]	valid_0's binary_logloss: 0.601447
[11]	valid_0's binary_logloss: 0.591179
[12]	valid_0's binary_logloss: 0.588372
[13]	valid_0's binary_logloss: 0.594197
[14]	valid_0's binary_logloss: 0.598823
[15]	valid_0's binary_logloss: 0.61018
[16]	valid_0's binary_logloss: 0.625001
[17]	valid_0's binary_logloss: 0.623898
[18]	valid_0's binary_logloss: 0.623476
[19]	valid_0's binary_logloss: 0.627288
[20]	valid_0's binary_logloss: 0.628713
[21]	valid_0's binary_logloss: 0.637802
[22]	valid_0's binary_logloss: 0.638965
[23]	valid_0's bi

[262]	valid_0's binary_logloss: 1.45906
[263]	valid_0's binary_logloss: 1.46166
[264]	valid_0's binary_logloss: 1.46358
[265]	valid_0's binary_logloss: 1.46405
[266]	valid_0's binary_logloss: 1.4657
[267]	valid_0's binary_logloss: 1.46529
[268]	valid_0's binary_logloss: 1.4687
[269]	valid_0's binary_logloss: 1.47146
[270]	valid_0's binary_logloss: 1.47196
[271]	valid_0's binary_logloss: 1.47291
[272]	valid_0's binary_logloss: 1.47318
[273]	valid_0's binary_logloss: 1.47537
[274]	valid_0's binary_logloss: 1.47881
[275]	valid_0's binary_logloss: 1.48042
[276]	valid_0's binary_logloss: 1.48187
[277]	valid_0's binary_logloss: 1.48183
[278]	valid_0's binary_logloss: 1.48165
[279]	valid_0's binary_logloss: 1.48366
[280]	valid_0's binary_logloss: 1.48454
[281]	valid_0's binary_logloss: 1.49103
[282]	valid_0's binary_logloss: 1.49498
[283]	valid_0's binary_logloss: 1.49392
[284]	valid_0's binary_logloss: 1.49418
[285]	valid_0's binary_logloss: 1.49455
[286]	valid_0's binary_logloss: 1.49351
[2

[601]	valid_0's binary_logloss: 1.63224
[602]	valid_0's binary_logloss: 1.63265
[603]	valid_0's binary_logloss: 1.63246
[604]	valid_0's binary_logloss: 1.63259
[605]	valid_0's binary_logloss: 1.63254
[606]	valid_0's binary_logloss: 1.6327
[607]	valid_0's binary_logloss: 1.63311
[608]	valid_0's binary_logloss: 1.63329
[609]	valid_0's binary_logloss: 1.63365
[610]	valid_0's binary_logloss: 1.63373
[611]	valid_0's binary_logloss: 1.63432
[612]	valid_0's binary_logloss: 1.63447
[613]	valid_0's binary_logloss: 1.63447
[614]	valid_0's binary_logloss: 1.63453
[615]	valid_0's binary_logloss: 1.63514
[616]	valid_0's binary_logloss: 1.63534
[617]	valid_0's binary_logloss: 1.63557
[618]	valid_0's binary_logloss: 1.63594
[619]	valid_0's binary_logloss: 1.63641
[620]	valid_0's binary_logloss: 1.63674
[621]	valid_0's binary_logloss: 1.63691
[622]	valid_0's binary_logloss: 1.63699
[623]	valid_0's binary_logloss: 1.63694
[624]	valid_0's binary_logloss: 1.63715
[625]	valid_0's binary_logloss: 1.63718
[

[1]	valid_0's binary_logloss: 0.591003
[2]	valid_0's binary_logloss: 0.586454
[3]	valid_0's binary_logloss: 0.583614
[4]	valid_0's binary_logloss: 0.592331
[5]	valid_0's binary_logloss: 0.59381
[6]	valid_0's binary_logloss: 0.586217
[7]	valid_0's binary_logloss: 0.583746
[8]	valid_0's binary_logloss: 0.597725
[9]	valid_0's binary_logloss: 0.596892
[10]	valid_0's binary_logloss: 0.601749
[11]	valid_0's binary_logloss: 0.600905
[12]	valid_0's binary_logloss: 0.602212
[13]	valid_0's binary_logloss: 0.610024
[14]	valid_0's binary_logloss: 0.613653
[15]	valid_0's binary_logloss: 0.619287
[16]	valid_0's binary_logloss: 0.617051
[17]	valid_0's binary_logloss: 0.622886
[18]	valid_0's binary_logloss: 0.630796
[19]	valid_0's binary_logloss: 0.633073
[20]	valid_0's binary_logloss: 0.63803
[21]	valid_0's binary_logloss: 0.643292
[22]	valid_0's binary_logloss: 0.645416
[23]	valid_0's binary_logloss: 0.647662
[24]	valid_0's binary_logloss: 0.647237
[25]	valid_0's binary_logloss: 0.65171
[26]	valid_0

[74]	valid_0's binary_logloss: 0.618757
[75]	valid_0's binary_logloss: 0.618636
[76]	valid_0's binary_logloss: 0.61852
[77]	valid_0's binary_logloss: 0.618401
[78]	valid_0's binary_logloss: 0.618287
[79]	valid_0's binary_logloss: 0.618169
[80]	valid_0's binary_logloss: 0.618056
[81]	valid_0's binary_logloss: 0.61794
[82]	valid_0's binary_logloss: 0.617829
[83]	valid_0's binary_logloss: 0.617714
[84]	valid_0's binary_logloss: 0.617604
[85]	valid_0's binary_logloss: 0.617491
[86]	valid_0's binary_logloss: 0.617383
[87]	valid_0's binary_logloss: 0.617271
[88]	valid_0's binary_logloss: 0.617164
[89]	valid_0's binary_logloss: 0.617053
[90]	valid_0's binary_logloss: 0.616948
[91]	valid_0's binary_logloss: 0.616839
[92]	valid_0's binary_logloss: 0.616735
[93]	valid_0's binary_logloss: 0.616628
[94]	valid_0's binary_logloss: 0.616525
[95]	valid_0's binary_logloss: 0.616419
[96]	valid_0's binary_logloss: 0.616318
[97]	valid_0's binary_logloss: 0.616213
[98]	valid_0's binary_logloss: 0.616113
[9

[379]	valid_0's binary_logloss: 0.594242
[380]	valid_0's binary_logloss: 0.594223
[381]	valid_0's binary_logloss: 0.59412
[382]	valid_0's binary_logloss: 0.594057
[383]	valid_0's binary_logloss: 0.593943
[384]	valid_0's binary_logloss: 0.593879
[385]	valid_0's binary_logloss: 0.593861
[386]	valid_0's binary_logloss: 0.59376
[387]	valid_0's binary_logloss: 0.593697
[388]	valid_0's binary_logloss: 0.593705
[389]	valid_0's binary_logloss: 0.593588
[390]	valid_0's binary_logloss: 0.593526
[391]	valid_0's binary_logloss: 0.593534
[392]	valid_0's binary_logloss: 0.593506
[393]	valid_0's binary_logloss: 0.593406
[394]	valid_0's binary_logloss: 0.593344
[395]	valid_0's binary_logloss: 0.593229
[396]	valid_0's binary_logloss: 0.593238
[397]	valid_0's binary_logloss: 0.593177
[398]	valid_0's binary_logloss: 0.593062
[399]	valid_0's binary_logloss: 0.593071
[400]	valid_0's binary_logloss: 0.59301
[401]	valid_0's binary_logloss: 0.592897
[402]	valid_0's binary_logloss: 0.592906
[403]	valid_0's bin

[714]	valid_0's binary_logloss: 0.588143
[715]	valid_0's binary_logloss: 0.588182
[716]	valid_0's binary_logloss: 0.588197
[717]	valid_0's binary_logloss: 0.588177
[718]	valid_0's binary_logloss: 0.588227
[719]	valid_0's binary_logloss: 0.588205
[720]	valid_0's binary_logloss: 0.588221
[721]	valid_0's binary_logloss: 0.588149
[722]	valid_0's binary_logloss: 0.588164
[723]	valid_0's binary_logloss: 0.588114
[724]	valid_0's binary_logloss: 0.588079
[725]	valid_0's binary_logloss: 0.588098
[726]	valid_0's binary_logloss: 0.588088
[727]	valid_0's binary_logloss: 0.588121
[728]	valid_0's binary_logloss: 0.588136
[729]	valid_0's binary_logloss: 0.588065
[730]	valid_0's binary_logloss: 0.588044
[731]	valid_0's binary_logloss: 0.588094
[732]	valid_0's binary_logloss: 0.588084
[733]	valid_0's binary_logloss: 0.588013
[734]	valid_0's binary_logloss: 0.588033
[735]	valid_0's binary_logloss: 0.58805
[736]	valid_0's binary_logloss: 0.588029
[737]	valid_0's binary_logloss: 0.588044
[738]	valid_0's b

[LightGBM] [Warning] Unknown parameter: eval_metric
[1]	valid_0's binary_logloss: 0.62765
[2]	valid_0's binary_logloss: 0.625194
[3]	valid_0's binary_logloss: 0.622856
[4]	valid_0's binary_logloss: 0.620622
[5]	valid_0's binary_logloss: 0.618419
[6]	valid_0's binary_logloss: 0.616328
[7]	valid_0's binary_logloss: 0.614372
[8]	valid_0's binary_logloss: 0.612426
[9]	valid_0's binary_logloss: 0.610673
[10]	valid_0's binary_logloss: 0.609095
[11]	valid_0's binary_logloss: 0.60775
[12]	valid_0's binary_logloss: 0.60622
[13]	valid_0's binary_logloss: 0.604325
[14]	valid_0's binary_logloss: 0.603207
[15]	valid_0's binary_logloss: 0.602013
[16]	valid_0's binary_logloss: 0.600227
[17]	valid_0's binary_logloss: 0.598514
[18]	valid_0's binary_logloss: 0.597334
[19]	valid_0's binary_logloss: 0.596281
[20]	valid_0's binary_logloss: 0.595171
[21]	valid_0's binary_logloss: 0.594073
[22]	valid_0's binary_logloss: 0.59313
[23]	valid_0's binary_logloss: 0.592506
[24]	valid_0's binary_logloss: 0.591405
[

[224]	valid_0's binary_logloss: 0.577299
[225]	valid_0's binary_logloss: 0.577364
[226]	valid_0's binary_logloss: 0.577764
[227]	valid_0's binary_logloss: 0.577816
[228]	valid_0's binary_logloss: 0.57823
[229]	valid_0's binary_logloss: 0.578208
[230]	valid_0's binary_logloss: 0.578621
[231]	valid_0's binary_logloss: 0.578998
[232]	valid_0's binary_logloss: 0.579648
[233]	valid_0's binary_logloss: 0.579872
[234]	valid_0's binary_logloss: 0.580302
[235]	valid_0's binary_logloss: 0.580522
[236]	valid_0's binary_logloss: 0.580702
[237]	valid_0's binary_logloss: 0.581197
[238]	valid_0's binary_logloss: 0.581388
[239]	valid_0's binary_logloss: 0.581596
[240]	valid_0's binary_logloss: 0.581925
[241]	valid_0's binary_logloss: 0.582124
[242]	valid_0's binary_logloss: 0.581848
[243]	valid_0's binary_logloss: 0.582153
[244]	valid_0's binary_logloss: 0.58188
[245]	valid_0's binary_logloss: 0.581987
[246]	valid_0's binary_logloss: 0.581912
[247]	valid_0's binary_logloss: 0.582032
[248]	valid_0's bi

[207]	valid_0's binary_logloss: 0.713049
[208]	valid_0's binary_logloss: 0.713733
[209]	valid_0's binary_logloss: 0.715167
[210]	valid_0's binary_logloss: 0.717021
[211]	valid_0's binary_logloss: 0.717774
[212]	valid_0's binary_logloss: 0.717477
[213]	valid_0's binary_logloss: 0.718389
[214]	valid_0's binary_logloss: 0.719968
[215]	valid_0's binary_logloss: 0.721407
[216]	valid_0's binary_logloss: 0.722799
[217]	valid_0's binary_logloss: 0.724525
[218]	valid_0's binary_logloss: 0.727346
[219]	valid_0's binary_logloss: 0.729514
[220]	valid_0's binary_logloss: 0.732092
[221]	valid_0's binary_logloss: 0.733573
[222]	valid_0's binary_logloss: 0.73406
[223]	valid_0's binary_logloss: 0.734764
[224]	valid_0's binary_logloss: 0.734915
[225]	valid_0's binary_logloss: 0.735908
[226]	valid_0's binary_logloss: 0.735819
[227]	valid_0's binary_logloss: 0.735479
[228]	valid_0's binary_logloss: 0.735515
[229]	valid_0's binary_logloss: 0.73536
[230]	valid_0's binary_logloss: 0.735462
[231]	valid_0's bi

[430]	valid_0's binary_logloss: 0.909783
[431]	valid_0's binary_logloss: 0.91005
[432]	valid_0's binary_logloss: 0.91107
[433]	valid_0's binary_logloss: 0.911723
[434]	valid_0's binary_logloss: 0.912234
[435]	valid_0's binary_logloss: 0.91344
[436]	valid_0's binary_logloss: 0.91465
[437]	valid_0's binary_logloss: 0.915761
[438]	valid_0's binary_logloss: 0.918377
[439]	valid_0's binary_logloss: 0.918977
[440]	valid_0's binary_logloss: 0.919823
[441]	valid_0's binary_logloss: 0.921603
[442]	valid_0's binary_logloss: 0.92156
[443]	valid_0's binary_logloss: 0.923589
[444]	valid_0's binary_logloss: 0.923935
[445]	valid_0's binary_logloss: 0.924964
[446]	valid_0's binary_logloss: 0.926726
[447]	valid_0's binary_logloss: 0.927614
[448]	valid_0's binary_logloss: 0.926733
[449]	valid_0's binary_logloss: 0.926873
[450]	valid_0's binary_logloss: 0.928691
[451]	valid_0's binary_logloss: 0.931054
[452]	valid_0's binary_logloss: 0.93216
[453]	valid_0's binary_logloss: 0.932336
[454]	valid_0's binary

[883]	valid_0's binary_logloss: 1.19496
[884]	valid_0's binary_logloss: 1.1957
[885]	valid_0's binary_logloss: 1.19596
[886]	valid_0's binary_logloss: 1.19717
[887]	valid_0's binary_logloss: 1.19773
[888]	valid_0's binary_logloss: 1.19809
[889]	valid_0's binary_logloss: 1.19915
[890]	valid_0's binary_logloss: 1.20016
[891]	valid_0's binary_logloss: 1.20206
[892]	valid_0's binary_logloss: 1.20344
[893]	valid_0's binary_logloss: 1.20282
[894]	valid_0's binary_logloss: 1.2041
[895]	valid_0's binary_logloss: 1.20391
[896]	valid_0's binary_logloss: 1.20358
[897]	valid_0's binary_logloss: 1.20392
[898]	valid_0's binary_logloss: 1.20479
[899]	valid_0's binary_logloss: 1.20524
[900]	valid_0's binary_logloss: 1.20521
[901]	valid_0's binary_logloss: 1.20598
[902]	valid_0's binary_logloss: 1.20585
[903]	valid_0's binary_logloss: 1.20587
[904]	valid_0's binary_logloss: 1.20641
[905]	valid_0's binary_logloss: 1.20721
[906]	valid_0's binary_logloss: 1.20712
[907]	valid_0's binary_logloss: 1.20719
[9

[211]	valid_0's binary_logloss: 0.661566
[212]	valid_0's binary_logloss: 0.664286
[213]	valid_0's binary_logloss: 0.664316
[214]	valid_0's binary_logloss: 0.662443
[215]	valid_0's binary_logloss: 0.662181
[216]	valid_0's binary_logloss: 0.661751
[217]	valid_0's binary_logloss: 0.663168
[218]	valid_0's binary_logloss: 0.664182
[219]	valid_0's binary_logloss: 0.664367
[220]	valid_0's binary_logloss: 0.665543
[221]	valid_0's binary_logloss: 0.66665
[222]	valid_0's binary_logloss: 0.667653
[223]	valid_0's binary_logloss: 0.668129
[224]	valid_0's binary_logloss: 0.67033
[225]	valid_0's binary_logloss: 0.670388
[226]	valid_0's binary_logloss: 0.670448
[227]	valid_0's binary_logloss: 0.671749
[228]	valid_0's binary_logloss: 0.671781
[229]	valid_0's binary_logloss: 0.671699
[230]	valid_0's binary_logloss: 0.67365
[231]	valid_0's binary_logloss: 0.674248
[232]	valid_0's binary_logloss: 0.673175
[233]	valid_0's binary_logloss: 0.672744
[234]	valid_0's binary_logloss: 0.674556
[235]	valid_0's bin

[563]	valid_0's binary_logloss: 0.899345
[564]	valid_0's binary_logloss: 0.900146
[565]	valid_0's binary_logloss: 0.900403
[566]	valid_0's binary_logloss: 0.900827
[567]	valid_0's binary_logloss: 0.90117
[568]	valid_0's binary_logloss: 0.901804
[569]	valid_0's binary_logloss: 0.902235
[570]	valid_0's binary_logloss: 0.903785
[571]	valid_0's binary_logloss: 0.904003
[572]	valid_0's binary_logloss: 0.903389
[573]	valid_0's binary_logloss: 0.90387
[574]	valid_0's binary_logloss: 0.904571
[575]	valid_0's binary_logloss: 0.904503
[576]	valid_0's binary_logloss: 0.905776
[577]	valid_0's binary_logloss: 0.905939
[578]	valid_0's binary_logloss: 0.905773
[579]	valid_0's binary_logloss: 0.908856
[580]	valid_0's binary_logloss: 0.908829
[581]	valid_0's binary_logloss: 0.908677
[582]	valid_0's binary_logloss: 0.909888
[583]	valid_0's binary_logloss: 0.911123
[584]	valid_0's binary_logloss: 0.911064
[585]	valid_0's binary_logloss: 0.914216
[586]	valid_0's binary_logloss: 0.914496
[587]	valid_0's bi

[150]	valid_0's binary_logloss: 1.00157
[151]	valid_0's binary_logloss: 1.00364
[152]	valid_0's binary_logloss: 1.00177
[153]	valid_0's binary_logloss: 1.00497
[154]	valid_0's binary_logloss: 1.00848
[155]	valid_0's binary_logloss: 1.01415
[156]	valid_0's binary_logloss: 1.01486
[157]	valid_0's binary_logloss: 1.01594
[158]	valid_0's binary_logloss: 1.01527
[159]	valid_0's binary_logloss: 1.01718
[160]	valid_0's binary_logloss: 1.01718
[161]	valid_0's binary_logloss: 1.02026
[162]	valid_0's binary_logloss: 1.02558
[163]	valid_0's binary_logloss: 1.02593
[164]	valid_0's binary_logloss: 1.02833
[165]	valid_0's binary_logloss: 1.03021
[166]	valid_0's binary_logloss: 1.03004
[167]	valid_0's binary_logloss: 1.03155
[168]	valid_0's binary_logloss: 1.03462
[169]	valid_0's binary_logloss: 1.0361
[170]	valid_0's binary_logloss: 1.03832
[171]	valid_0's binary_logloss: 1.03768
[172]	valid_0's binary_logloss: 1.04179
[173]	valid_0's binary_logloss: 1.04662
[174]	valid_0's binary_logloss: 1.04743
[

[479]	valid_0's binary_logloss: 1.44085
[480]	valid_0's binary_logloss: 1.44097
[481]	valid_0's binary_logloss: 1.44111
[482]	valid_0's binary_logloss: 1.44256
[483]	valid_0's binary_logloss: 1.442
[484]	valid_0's binary_logloss: 1.44385
[485]	valid_0's binary_logloss: 1.44555
[486]	valid_0's binary_logloss: 1.44771
[487]	valid_0's binary_logloss: 1.44832
[488]	valid_0's binary_logloss: 1.44842
[489]	valid_0's binary_logloss: 1.44818
[490]	valid_0's binary_logloss: 1.44952
[491]	valid_0's binary_logloss: 1.44959
[492]	valid_0's binary_logloss: 1.44964
[493]	valid_0's binary_logloss: 1.4495
[494]	valid_0's binary_logloss: 1.4495
[495]	valid_0's binary_logloss: 1.44965
[496]	valid_0's binary_logloss: 1.4499
[497]	valid_0's binary_logloss: 1.45179
[498]	valid_0's binary_logloss: 1.45242
[499]	valid_0's binary_logloss: 1.45288
[500]	valid_0's binary_logloss: 1.45798
[501]	valid_0's binary_logloss: 1.45853
[502]	valid_0's binary_logloss: 1.45867
[503]	valid_0's binary_logloss: 1.45955
[504]

[159]	valid_0's binary_logloss: 1.00152
[160]	valid_0's binary_logloss: 1.00694
[161]	valid_0's binary_logloss: 1.01127
[162]	valid_0's binary_logloss: 1.0117
[163]	valid_0's binary_logloss: 1.01067
[164]	valid_0's binary_logloss: 1.0148
[165]	valid_0's binary_logloss: 1.01496
[166]	valid_0's binary_logloss: 1.01689
[167]	valid_0's binary_logloss: 1.01785
[168]	valid_0's binary_logloss: 1.01958
[169]	valid_0's binary_logloss: 1.02225
[170]	valid_0's binary_logloss: 1.02706
[171]	valid_0's binary_logloss: 1.02648
[172]	valid_0's binary_logloss: 1.03013
[173]	valid_0's binary_logloss: 1.03036
[174]	valid_0's binary_logloss: 1.03078
[175]	valid_0's binary_logloss: 1.03273
[176]	valid_0's binary_logloss: 1.03224
[177]	valid_0's binary_logloss: 1.03706
[178]	valid_0's binary_logloss: 1.04213
[179]	valid_0's binary_logloss: 1.04551
[180]	valid_0's binary_logloss: 1.0528
[181]	valid_0's binary_logloss: 1.05508
[182]	valid_0's binary_logloss: 1.05898
[183]	valid_0's binary_logloss: 1.06063
[18

[372]	valid_0's binary_logloss: 1.3615
[373]	valid_0's binary_logloss: 1.36342
[374]	valid_0's binary_logloss: 1.36675
[375]	valid_0's binary_logloss: 1.36565
[376]	valid_0's binary_logloss: 1.36601
[377]	valid_0's binary_logloss: 1.36671
[378]	valid_0's binary_logloss: 1.36864
[379]	valid_0's binary_logloss: 1.36769
[380]	valid_0's binary_logloss: 1.36955
[381]	valid_0's binary_logloss: 1.37149
[382]	valid_0's binary_logloss: 1.37299
[383]	valid_0's binary_logloss: 1.37331
[384]	valid_0's binary_logloss: 1.37421
[385]	valid_0's binary_logloss: 1.37539
[386]	valid_0's binary_logloss: 1.37774
[387]	valid_0's binary_logloss: 1.38087
[388]	valid_0's binary_logloss: 1.38153
[389]	valid_0's binary_logloss: 1.38237
[390]	valid_0's binary_logloss: 1.38444
[391]	valid_0's binary_logloss: 1.39174
[392]	valid_0's binary_logloss: 1.39255
[393]	valid_0's binary_logloss: 1.39336
[394]	valid_0's binary_logloss: 1.39279
[395]	valid_0's binary_logloss: 1.39335
[396]	valid_0's binary_logloss: 1.39624
[

[142]	valid_0's binary_logloss: 0.911562
[143]	valid_0's binary_logloss: 0.912398
[144]	valid_0's binary_logloss: 0.913811
[145]	valid_0's binary_logloss: 0.912472
[146]	valid_0's binary_logloss: 0.915981
[147]	valid_0's binary_logloss: 0.917239
[148]	valid_0's binary_logloss: 0.918342
[149]	valid_0's binary_logloss: 0.921311
[150]	valid_0's binary_logloss: 0.925835
[151]	valid_0's binary_logloss: 0.928481
[152]	valid_0's binary_logloss: 0.930854
[153]	valid_0's binary_logloss: 0.932682
[154]	valid_0's binary_logloss: 0.936728
[155]	valid_0's binary_logloss: 0.939282
[156]	valid_0's binary_logloss: 0.94043
[157]	valid_0's binary_logloss: 0.940822
[158]	valid_0's binary_logloss: 0.947517
[159]	valid_0's binary_logloss: 0.953031
[160]	valid_0's binary_logloss: 0.957383
[161]	valid_0's binary_logloss: 0.958418
[162]	valid_0's binary_logloss: 0.960118
[163]	valid_0's binary_logloss: 0.961274
[164]	valid_0's binary_logloss: 0.965883
[165]	valid_0's binary_logloss: 0.966682
[166]	valid_0's b

| 18        | 0.4655    | 0.1359    | 8.292     | 479.0     |
[LightGBM] [Warning] Unknown parameter: eval_metric
[1]	valid_0's binary_logloss: 0.62861
[2]	valid_0's binary_logloss: 0.627144
[3]	valid_0's binary_logloss: 0.625752
[4]	valid_0's binary_logloss: 0.624432
[5]	valid_0's binary_logloss: 0.62318
[6]	valid_0's binary_logloss: 0.621993
[7]	valid_0's binary_logloss: 0.620906
[8]	valid_0's binary_logloss: 0.619834
[9]	valid_0's binary_logloss: 0.618855
[10]	valid_0's binary_logloss: 0.617889
[11]	valid_0's binary_logloss: 0.61701
[12]	valid_0's binary_logloss: 0.616141
[13]	valid_0's binary_logloss: 0.615354
[14]	valid_0's binary_logloss: 0.614613
[15]	valid_0's binary_logloss: 0.614033
[16]	valid_0's binary_logloss: 0.613352
[17]	valid_0's binary_logloss: 0.612794
[18]	valid_0's binary_logloss: 0.612191
[19]	valid_0's binary_logloss: 0.611704
[20]	valid_0's binary_logloss: 0.611172
[21]	valid_0's binary_logloss: 0.610225
[22]	valid_0's binary_logloss: 0.609579
[23]	valid_0's bin

[1]	valid_0's binary_logloss: 0.619011
[2]	valid_0's binary_logloss: 0.612399
[3]	valid_0's binary_logloss: 0.609543
[4]	valid_0's binary_logloss: 0.603239
[5]	valid_0's binary_logloss: 0.596414
[6]	valid_0's binary_logloss: 0.591718
[7]	valid_0's binary_logloss: 0.59087
[8]	valid_0's binary_logloss: 0.58442
[9]	valid_0's binary_logloss: 0.581857
[10]	valid_0's binary_logloss: 0.582719
[11]	valid_0's binary_logloss: 0.582956
[12]	valid_0's binary_logloss: 0.582711
[13]	valid_0's binary_logloss: 0.586101
[14]	valid_0's binary_logloss: 0.586035
[15]	valid_0's binary_logloss: 0.585045
[16]	valid_0's binary_logloss: 0.58807
[17]	valid_0's binary_logloss: 0.589694
[18]	valid_0's binary_logloss: 0.585489
[19]	valid_0's binary_logloss: 0.58606
[20]	valid_0's binary_logloss: 0.581435
[21]	valid_0's binary_logloss: 0.584043
[22]	valid_0's binary_logloss: 0.584265
[23]	valid_0's binary_logloss: 0.583011
[24]	valid_0's binary_logloss: 0.58381
[25]	valid_0's binary_logloss: 0.583789
[26]	valid_0's

[386]	valid_0's binary_logloss: 0.804088
[387]	valid_0's binary_logloss: 0.803878
[388]	valid_0's binary_logloss: 0.806383
[389]	valid_0's binary_logloss: 0.807419
[390]	valid_0's binary_logloss: 0.810842
[391]	valid_0's binary_logloss: 0.811467
[392]	valid_0's binary_logloss: 0.811338
[393]	valid_0's binary_logloss: 0.813374
[394]	valid_0's binary_logloss: 0.814302
[395]	valid_0's binary_logloss: 0.81599
[396]	valid_0's binary_logloss: 0.816413
[397]	valid_0's binary_logloss: 0.816843
[398]	valid_0's binary_logloss: 0.816145
[399]	valid_0's binary_logloss: 0.816936
[400]	valid_0's binary_logloss: 0.819037
[401]	valid_0's binary_logloss: 0.819729
[402]	valid_0's binary_logloss: 0.822005
[403]	valid_0's binary_logloss: 0.823695
[404]	valid_0's binary_logloss: 0.823536
[405]	valid_0's binary_logloss: 0.823515
[406]	valid_0's binary_logloss: 0.823026
[407]	valid_0's binary_logloss: 0.822993
[408]	valid_0's binary_logloss: 0.82326
[409]	valid_0's binary_logloss: 0.823717
[410]	valid_0's bi

[735]	valid_0's binary_logloss: 1.04333
[736]	valid_0's binary_logloss: 1.04404
[737]	valid_0's binary_logloss: 1.04526
[738]	valid_0's binary_logloss: 1.04776
[739]	valid_0's binary_logloss: 1.04882
[740]	valid_0's binary_logloss: 1.04904
[741]	valid_0's binary_logloss: 1.04962
[742]	valid_0's binary_logloss: 1.04947
[743]	valid_0's binary_logloss: 1.05045
[744]	valid_0's binary_logloss: 1.05117
[745]	valid_0's binary_logloss: 1.05151
[746]	valid_0's binary_logloss: 1.05184
[747]	valid_0's binary_logloss: 1.05197
[748]	valid_0's binary_logloss: 1.05326
[749]	valid_0's binary_logloss: 1.05411
[750]	valid_0's binary_logloss: 1.05451
[751]	valid_0's binary_logloss: 1.05485
[752]	valid_0's binary_logloss: 1.05625
[753]	valid_0's binary_logloss: 1.05645
[754]	valid_0's binary_logloss: 1.05642
[755]	valid_0's binary_logloss: 1.0566
[756]	valid_0's binary_logloss: 1.05628
[757]	valid_0's binary_logloss: 1.05619
[758]	valid_0's binary_logloss: 1.05601
[759]	valid_0's binary_logloss: 1.05645
[

[152]	valid_0's binary_logloss: 0.678977
[153]	valid_0's binary_logloss: 0.679079
[154]	valid_0's binary_logloss: 0.681025
[155]	valid_0's binary_logloss: 0.682824
[156]	valid_0's binary_logloss: 0.683522
[157]	valid_0's binary_logloss: 0.68307
[158]	valid_0's binary_logloss: 0.683851
[159]	valid_0's binary_logloss: 0.685482
[160]	valid_0's binary_logloss: 0.686047
[161]	valid_0's binary_logloss: 0.688098
[162]	valid_0's binary_logloss: 0.688356
[163]	valid_0's binary_logloss: 0.690699
[164]	valid_0's binary_logloss: 0.69062
[165]	valid_0's binary_logloss: 0.691812
[166]	valid_0's binary_logloss: 0.693132
[167]	valid_0's binary_logloss: 0.693658
[168]	valid_0's binary_logloss: 0.69644
[169]	valid_0's binary_logloss: 0.696819
[170]	valid_0's binary_logloss: 0.69764
[171]	valid_0's binary_logloss: 0.697056
[172]	valid_0's binary_logloss: 0.697875
[173]	valid_0's binary_logloss: 0.699022
[174]	valid_0's binary_logloss: 0.69997
[175]	valid_0's binary_logloss: 0.702974
[176]	valid_0's binar

[371]	valid_0's binary_logloss: 0.909394
[372]	valid_0's binary_logloss: 0.911139
[373]	valid_0's binary_logloss: 0.912302
[374]	valid_0's binary_logloss: 0.913423
[375]	valid_0's binary_logloss: 0.912979
[376]	valid_0's binary_logloss: 0.915103
[377]	valid_0's binary_logloss: 0.91601
[378]	valid_0's binary_logloss: 0.916327
[379]	valid_0's binary_logloss: 0.917016
[380]	valid_0's binary_logloss: 0.916916
[381]	valid_0's binary_logloss: 0.91814
[382]	valid_0's binary_logloss: 0.920515
[383]	valid_0's binary_logloss: 0.920119
[384]	valid_0's binary_logloss: 0.921804
[385]	valid_0's binary_logloss: 0.923051
[386]	valid_0's binary_logloss: 0.923725
[387]	valid_0's binary_logloss: 0.924604
[388]	valid_0's binary_logloss: 0.926589
[389]	valid_0's binary_logloss: 0.927641
[390]	valid_0's binary_logloss: 0.929816
[391]	valid_0's binary_logloss: 0.932301
[392]	valid_0's binary_logloss: 0.933339
[393]	valid_0's binary_logloss: 0.935067
[394]	valid_0's binary_logloss: 0.934498
[395]	valid_0's bi

[643]	valid_0's binary_logloss: 1.13914
[644]	valid_0's binary_logloss: 1.13895
[645]	valid_0's binary_logloss: 1.13841
[646]	valid_0's binary_logloss: 1.13874
[647]	valid_0's binary_logloss: 1.1398
[648]	valid_0's binary_logloss: 1.14106
[649]	valid_0's binary_logloss: 1.14062
[650]	valid_0's binary_logloss: 1.14132
[651]	valid_0's binary_logloss: 1.14249
[652]	valid_0's binary_logloss: 1.1427
[653]	valid_0's binary_logloss: 1.14326
[654]	valid_0's binary_logloss: 1.14501
[655]	valid_0's binary_logloss: 1.14496
[656]	valid_0's binary_logloss: 1.14426
[657]	valid_0's binary_logloss: 1.14461
[658]	valid_0's binary_logloss: 1.14551
[659]	valid_0's binary_logloss: 1.14523
[660]	valid_0's binary_logloss: 1.14541
[661]	valid_0's binary_logloss: 1.14585
[662]	valid_0's binary_logloss: 1.14635
[663]	valid_0's binary_logloss: 1.14679
[664]	valid_0's binary_logloss: 1.14805
[665]	valid_0's binary_logloss: 1.14975
[666]	valid_0's binary_logloss: 1.14961
[667]	valid_0's binary_logloss: 1.15014
[6

[154]	valid_0's binary_logloss: 1.08588
[155]	valid_0's binary_logloss: 1.08339
[156]	valid_0's binary_logloss: 1.08497
[157]	valid_0's binary_logloss: 1.08304
[158]	valid_0's binary_logloss: 1.08597
[159]	valid_0's binary_logloss: 1.08948
[160]	valid_0's binary_logloss: 1.09344
[161]	valid_0's binary_logloss: 1.10023
[162]	valid_0's binary_logloss: 1.10218
[163]	valid_0's binary_logloss: 1.10184
[164]	valid_0's binary_logloss: 1.10415
[165]	valid_0's binary_logloss: 1.10503
[166]	valid_0's binary_logloss: 1.10523
[167]	valid_0's binary_logloss: 1.10793
[168]	valid_0's binary_logloss: 1.1111
[169]	valid_0's binary_logloss: 1.11236
[170]	valid_0's binary_logloss: 1.12252
[171]	valid_0's binary_logloss: 1.1266
[172]	valid_0's binary_logloss: 1.12809
[173]	valid_0's binary_logloss: 1.12906
[174]	valid_0's binary_logloss: 1.13209
[175]	valid_0's binary_logloss: 1.13295
[176]	valid_0's binary_logloss: 1.1363
[177]	valid_0's binary_logloss: 1.13687
[178]	valid_0's binary_logloss: 1.13692
[17

[69]	valid_0's binary_logloss: 0.714057
[70]	valid_0's binary_logloss: 0.719043
[71]	valid_0's binary_logloss: 0.723369
[72]	valid_0's binary_logloss: 0.724116
[73]	valid_0's binary_logloss: 0.727088
[74]	valid_0's binary_logloss: 0.727431
[75]	valid_0's binary_logloss: 0.72814
[76]	valid_0's binary_logloss: 0.727692
[77]	valid_0's binary_logloss: 0.731299
[78]	valid_0's binary_logloss: 0.73433
[79]	valid_0's binary_logloss: 0.735301
[80]	valid_0's binary_logloss: 0.734999
[81]	valid_0's binary_logloss: 0.735263
[82]	valid_0's binary_logloss: 0.735914
[83]	valid_0's binary_logloss: 0.740757
[84]	valid_0's binary_logloss: 0.74114
[85]	valid_0's binary_logloss: 0.746117
[86]	valid_0's binary_logloss: 0.748098
[87]	valid_0's binary_logloss: 0.752214
[88]	valid_0's binary_logloss: 0.751365
[89]	valid_0's binary_logloss: 0.753343
[90]	valid_0's binary_logloss: 0.759533
[91]	valid_0's binary_logloss: 0.761579
[92]	valid_0's binary_logloss: 0.765469
| 23        | 0.5357    | 0.1323    | 7.854

[206]	valid_0's binary_logloss: 0.630123
[207]	valid_0's binary_logloss: 0.630571
[208]	valid_0's binary_logloss: 0.628282
[209]	valid_0's binary_logloss: 0.628134
[210]	valid_0's binary_logloss: 0.628227
[211]	valid_0's binary_logloss: 0.627429
[212]	valid_0's binary_logloss: 0.629849
[213]	valid_0's binary_logloss: 0.628894
[214]	valid_0's binary_logloss: 0.626209
[215]	valid_0's binary_logloss: 0.627342
[216]	valid_0's binary_logloss: 0.628021
[217]	valid_0's binary_logloss: 0.629184
[218]	valid_0's binary_logloss: 0.629758
[219]	valid_0's binary_logloss: 0.628838
[220]	valid_0's binary_logloss: 0.629231
[221]	valid_0's binary_logloss: 0.630973
[222]	valid_0's binary_logloss: 0.628576
[223]	valid_0's binary_logloss: 0.629372
[224]	valid_0's binary_logloss: 0.631951
[225]	valid_0's binary_logloss: 0.634784
[226]	valid_0's binary_logloss: 0.636335
[227]	valid_0's binary_logloss: 0.635748
[228]	valid_0's binary_logloss: 0.635338
[229]	valid_0's binary_logloss: 0.634369
[230]	valid_0's 

[LightGBM] [Warning] Unknown parameter: eval_metric
[1]	valid_0's binary_logloss: 0.62556
[2]	valid_0's binary_logloss: 0.621896
[3]	valid_0's binary_logloss: 0.61547
[4]	valid_0's binary_logloss: 0.61302
[5]	valid_0's binary_logloss: 0.608714
[6]	valid_0's binary_logloss: 0.604967
[7]	valid_0's binary_logloss: 0.60268
[8]	valid_0's binary_logloss: 0.598167
[9]	valid_0's binary_logloss: 0.598104
[10]	valid_0's binary_logloss: 0.598731
[11]	valid_0's binary_logloss: 0.598815
[12]	valid_0's binary_logloss: 0.594631
[13]	valid_0's binary_logloss: 0.593947
[14]	valid_0's binary_logloss: 0.592141
[15]	valid_0's binary_logloss: 0.593765
[16]	valid_0's binary_logloss: 0.593037
[17]	valid_0's binary_logloss: 0.591122
[18]	valid_0's binary_logloss: 0.591003
[19]	valid_0's binary_logloss: 0.591901
[20]	valid_0's binary_logloss: 0.590782
[21]	valid_0's binary_logloss: 0.590675
[22]	valid_0's binary_logloss: 0.591762
[23]	valid_0's binary_logloss: 0.590304
[24]	valid_0's binary_logloss: 0.590353
[

[421]	valid_0's binary_logloss: 0.606966
[422]	valid_0's binary_logloss: 0.606976
[423]	valid_0's binary_logloss: 0.607429
[424]	valid_0's binary_logloss: 0.607245
[425]	valid_0's binary_logloss: 0.60687
[426]	valid_0's binary_logloss: 0.60707
[427]	valid_0's binary_logloss: 0.606719
[428]	valid_0's binary_logloss: 0.606172
[429]	valid_0's binary_logloss: 0.607016
[430]	valid_0's binary_logloss: 0.606761
[431]	valid_0's binary_logloss: 0.607195
[432]	valid_0's binary_logloss: 0.606709
[433]	valid_0's binary_logloss: 0.606477
[434]	valid_0's binary_logloss: 0.606124
[435]	valid_0's binary_logloss: 0.606199
[436]	valid_0's binary_logloss: 0.606681
[437]	valid_0's binary_logloss: 0.607211
[438]	valid_0's binary_logloss: 0.6071
[439]	valid_0's binary_logloss: 0.607304
[440]	valid_0's binary_logloss: 0.607639
[441]	valid_0's binary_logloss: 0.60727
[442]	valid_0's binary_logloss: 0.607459
[443]	valid_0's binary_logloss: 0.607627
[444]	valid_0's binary_logloss: 0.607764
[445]	valid_0's binar

[316]	valid_0's binary_logloss: 0.589422
[317]	valid_0's binary_logloss: 0.589207
[318]	valid_0's binary_logloss: 0.588972
[319]	valid_0's binary_logloss: 0.589263
[320]	valid_0's binary_logloss: 0.589273
[321]	valid_0's binary_logloss: 0.589021
[322]	valid_0's binary_logloss: 0.588634
[323]	valid_0's binary_logloss: 0.588989
[324]	valid_0's binary_logloss: 0.589172
[325]	valid_0's binary_logloss: 0.589386
[326]	valid_0's binary_logloss: 0.589173
[327]	valid_0's binary_logloss: 0.589185
[328]	valid_0's binary_logloss: 0.589008
[329]	valid_0's binary_logloss: 0.588716
[330]	valid_0's binary_logloss: 0.58868
[331]	valid_0's binary_logloss: 0.588677
[332]	valid_0's binary_logloss: 0.588647
[333]	valid_0's binary_logloss: 0.588426
[334]	valid_0's binary_logloss: 0.588112
[335]	valid_0's binary_logloss: 0.588297
[336]	valid_0's binary_logloss: 0.588252
[337]	valid_0's binary_logloss: 0.58804
[338]	valid_0's binary_logloss: 0.588049
[339]	valid_0's binary_logloss: 0.58765
[340]	valid_0's bin

| 27        | 0.5133    | 0.02421   | 3.87      | 649.8     |
[LightGBM] [Warning] Unknown parameter: eval_metric
[1]	valid_0's binary_logloss: 0.628644
[2]	valid_0's binary_logloss: 0.62718
[3]	valid_0's binary_logloss: 0.625728
[4]	valid_0's binary_logloss: 0.624353
[5]	valid_0's binary_logloss: 0.622987
[6]	valid_0's binary_logloss: 0.621695
[7]	valid_0's binary_logloss: 0.62082
[8]	valid_0's binary_logloss: 0.619472
[9]	valid_0's binary_logloss: 0.618545
[10]	valid_0's binary_logloss: 0.6176
[11]	valid_0's binary_logloss: 0.616496
[12]	valid_0's binary_logloss: 0.615647
[13]	valid_0's binary_logloss: 0.614928
[14]	valid_0's binary_logloss: 0.614202
[15]	valid_0's binary_logloss: 0.613026
[16]	valid_0's binary_logloss: 0.612352
[17]	valid_0's binary_logloss: 0.611591
[18]	valid_0's binary_logloss: 0.61088
[19]	valid_0's binary_logloss: 0.610098
[20]	valid_0's binary_logloss: 0.609189
[21]	valid_0's binary_logloss: 0.608541
[22]	valid_0's binary_logloss: 0.607901
[23]	valid_0's binar

[336]	valid_0's binary_logloss: 0.572509
[337]	valid_0's binary_logloss: 0.572518
[338]	valid_0's binary_logloss: 0.572449
[339]	valid_0's binary_logloss: 0.57254
[340]	valid_0's binary_logloss: 0.572553
[341]	valid_0's binary_logloss: 0.572693
[342]	valid_0's binary_logloss: 0.572864
[343]	valid_0's binary_logloss: 0.572911
[344]	valid_0's binary_logloss: 0.573131
[345]	valid_0's binary_logloss: 0.572875
[346]	valid_0's binary_logloss: 0.572934
[347]	valid_0's binary_logloss: 0.572793
[348]	valid_0's binary_logloss: 0.572965
[349]	valid_0's binary_logloss: 0.573102
[350]	valid_0's binary_logloss: 0.573327
[351]	valid_0's binary_logloss: 0.573077
[352]	valid_0's binary_logloss: 0.572982
[353]	valid_0's binary_logloss: 0.573237
[354]	valid_0's binary_logloss: 0.573282
[355]	valid_0's binary_logloss: 0.573422
[356]	valid_0's binary_logloss: 0.573404
[357]	valid_0's binary_logloss: 0.573295
[358]	valid_0's binary_logloss: 0.573356
[359]	valid_0's binary_logloss: 0.573612
[360]	valid_0's b

[624]	valid_0's binary_logloss: 0.580654
[625]	valid_0's binary_logloss: 0.580752
[626]	valid_0's binary_logloss: 0.580884
[627]	valid_0's binary_logloss: 0.580866
[628]	valid_0's binary_logloss: 0.580973
[629]	valid_0's binary_logloss: 0.580936
[630]	valid_0's binary_logloss: 0.580877
[631]	valid_0's binary_logloss: 0.580975
[632]	valid_0's binary_logloss: 0.581155
[633]	valid_0's binary_logloss: 0.581128
[634]	valid_0's binary_logloss: 0.581173
[635]	valid_0's binary_logloss: 0.581335
[636]	valid_0's binary_logloss: 0.581345
[637]	valid_0's binary_logloss: 0.58154
[638]	valid_0's binary_logloss: 0.581596
[639]	valid_0's binary_logloss: 0.581732
[640]	valid_0's binary_logloss: 0.582037
[641]	valid_0's binary_logloss: 0.581992
[642]	valid_0's binary_logloss: 0.582161
[643]	valid_0's binary_logloss: 0.582354
[644]	valid_0's binary_logloss: 0.582436
[645]	valid_0's binary_logloss: 0.582545
[646]	valid_0's binary_logloss: 0.58273
[647]	valid_0's binary_logloss: 0.582905
[648]	valid_0's bi

| 28        | 0.5556    | 0.01084   | 4.683     | 855.9     |
[LightGBM] [Warning] Unknown parameter: eval_metric
[1]	valid_0's binary_logloss: 0.614191
[2]	valid_0's binary_logloss: 0.607403
[3]	valid_0's binary_logloss: 0.601126
[4]	valid_0's binary_logloss: 0.596496
[5]	valid_0's binary_logloss: 0.591837
[6]	valid_0's binary_logloss: 0.589963
[7]	valid_0's binary_logloss: 0.585016
[8]	valid_0's binary_logloss: 0.582024
[9]	valid_0's binary_logloss: 0.577953
[10]	valid_0's binary_logloss: 0.578576
[11]	valid_0's binary_logloss: 0.580944
[12]	valid_0's binary_logloss: 0.575467
[13]	valid_0's binary_logloss: 0.576737
[14]	valid_0's binary_logloss: 0.578576
[15]	valid_0's binary_logloss: 0.57768
[16]	valid_0's binary_logloss: 0.582083
[17]	valid_0's binary_logloss: 0.584463
[18]	valid_0's binary_logloss: 0.584284
[19]	valid_0's binary_logloss: 0.58368
[20]	valid_0's binary_logloss: 0.585133
[21]	valid_0's binary_logloss: 0.588306
[22]	valid_0's binary_logloss: 0.585571
[23]	valid_0's bi

[396]	valid_0's binary_logloss: 0.99555
[397]	valid_0's binary_logloss: 0.995801
[398]	valid_0's binary_logloss: 0.995926
[399]	valid_0's binary_logloss: 0.999644
[400]	valid_0's binary_logloss: 0.999172
[401]	valid_0's binary_logloss: 1.00123
[402]	valid_0's binary_logloss: 1.00169
[403]	valid_0's binary_logloss: 1.00154
[404]	valid_0's binary_logloss: 1.00213
[405]	valid_0's binary_logloss: 1.00533
[406]	valid_0's binary_logloss: 1.00516
[407]	valid_0's binary_logloss: 1.00557
[408]	valid_0's binary_logloss: 1.00774
[409]	valid_0's binary_logloss: 1.00941
[410]	valid_0's binary_logloss: 1.01251
[411]	valid_0's binary_logloss: 1.01447
[412]	valid_0's binary_logloss: 1.01648
[413]	valid_0's binary_logloss: 1.01866
[414]	valid_0's binary_logloss: 1.02087
[415]	valid_0's binary_logloss: 1.02507
[416]	valid_0's binary_logloss: 1.02509
[417]	valid_0's binary_logloss: 1.0273
[418]	valid_0's binary_logloss: 1.02721
[419]	valid_0's binary_logloss: 1.02918
[420]	valid_0's binary_logloss: 1.031

[324]	valid_0's binary_logloss: 0.673046
[325]	valid_0's binary_logloss: 0.673749
[326]	valid_0's binary_logloss: 0.674151
[327]	valid_0's binary_logloss: 0.675673
[328]	valid_0's binary_logloss: 0.676132
[329]	valid_0's binary_logloss: 0.676481
[330]	valid_0's binary_logloss: 0.677112
[331]	valid_0's binary_logloss: 0.677536
[332]	valid_0's binary_logloss: 0.677728
| 30        | 0.5586    | 0.03207   | 6.263     | 332.2     |


In [35]:
best_hyperparameters = bayesian_opt.max
best_hyperparameters['params'] = {param: int(value) if param in [
    'depth', 'iterations', 'n_estimators'] else value for param, value in best_hyperparameters['params'].items()}
print("Best hyperparameters:", best_hyperparameters['params'])
print(f"Best R-squared Score: {best_hyperparameters['target']:.4f}")

Best hyperparameters: {'learning_rate': 0.012925143711798016, 'max_depth': 9.046486124621914, 'n_estimators': 339}
Best R-squared Score: 0.5893


In [36]:
params = {
    'max_depth': 9,            
    'learning_rate': 0.012925143711798016,  
    'n_estimators': 339
}

model_LGBM = LGBMClassifier(**params, random_state=27)
model_LGBM.fit(x_train_LGBM, y_train_LGBM)
# предсказание ответов для тренеровочной выборки
y_pred_LGBM_train = model_LGBM.predict(x_train_LGBM)
# предсказание ответов для тестовой выборки
y_pred_LGBM = model_LGBM.predict(x_valid_LGBM)

print("Train LGBM f1_score: ", f1_score(y_train_LGBM, y_pred_LGBM_train))
print("Train LGBM accuracy_score: ", accuracy_score(y_train_LGBM, y_pred_LGBM_train))
print("Validation LGBM f1_score: ", f1_score(y_valid_LGBM, y_pred_LGBM))
print("Validation LGBM accuracy_score: ", accuracy_score(y_valid_LGBM, y_pred_LGBM))

Train LGBM f1_score:  0.8949416342412452
Train LGBM accuracy_score:  0.9304123711340206
Validation LGBM f1_score:  0.5892857142857143
Validation LGBM accuracy_score:  0.764102564102564


После подбора гиперпараметров мы улучшили метрики и снизили переобучение для модели LGBMClassifier на вадидации f1_score:  0.5892, accuracy_score:  0.764.

**xgboost**

In [37]:
def XGB_cv(max_depth, learning_rate, n_estimators):
    max_depth = int(max_depth)
    n_estimators = int(n_estimators)
    model = xg.XGBClassifier(
        max_depth=max_depth,
        learning_rate=learning_rate,
        n_estimators=n_estimators,
        random_seed=27
    )
    model.fit(x_train_xgb, y_train_xgb)
    return f1_score(y_valid_xgb, model.predict(x_valid_xgb))

In [38]:
param_space = {
    'max_depth': (2, 10),            
    'learning_rate': (0.001, 0.3),  
    'n_estimators': (10, 2000)
}

bayesian_opt = BayesianOptimization(
    allow_duplicate_points=True,f=XGB_cv, pbounds=param_space, random_state=27)
bayesian_opt.maximize(init_points=5, n_iter=50)
results = pd.DataFrame(bayesian_opt.res)
results.sort_values(by='target', ascending=False, inplace=True)

|   iter    |  target   | learni... | max_depth | n_esti... |
-------------------------------------------------------------


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:20:51] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 1         | 0.5439    | 0.1283    | 8.517     | 1.473e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:20:56] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 2         | 0.5641    | 0.2605    | 5.067     | 1.959e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:20:59] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 3         | 0.5546    | 0.2681    | 3.678     | 1.486e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:21:00] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 4         | 0.547     | 0.1993    | 9.094     | 1.717e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:21:04] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 5         | 0.5614    | 0.225     | 8.961     | 381.6     |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:21:06] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 6         | 0.5517    | 0.1159    | 8.347     | 381.7     |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:21:07] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 7         | 0.5133    | 0.003044  | 6.786     | 591.6     |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:21:10] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 8         | 0.569     | 0.03185   | 4.821     | 1.958e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:21:14] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 9         | 0.5739    | 0.08527   | 6.445     | 1.957e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:21:18] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 10        | 0.5862    | 0.1789    | 5.826     | 1.955e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:21:21] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 11        | 0.5517    | 0.1552    | 7.168     | 1.952e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:21:25] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 12        | 0.547     | 0.1187    | 3.21      | 1.955e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:21:27] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 13        | 0.5517    | 0.1432    | 8.096     | 1.955e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:21:32] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 14        | 0.5614    | 0.2192    | 6.323     | 1.956e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:21:36] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 15        | 0.569     | 0.2074    | 4.879     | 1.959e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:21:38] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 16        | 0.5739    | 0.08469   | 5.496     | 1.955e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:21:42] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 17        | 0.569     | 0.2316    | 6.055     | 1.955e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:21:45] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 18        | 0.5641    | 0.2684    | 6.392     | 1.955e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:21:49] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 19        | 0.5517    | 0.1652    | 5.595     | 1.956e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:21:52] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 20        | 0.5641    | 0.1751    | 6.333     | 1.958e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:21:56] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 21        | 0.5882    | 0.2155    | 4.963     | 1.958e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:21:59] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 22        | 0.5378    | 0.29      | 4.488     | 1.958e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:01] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 23        | 0.5517    | 0.2569    | 4.747     | 1.959e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:04] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 24        | 0.5593    | 0.1546    | 4.816     | 1.958e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:07] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 25        | 0.5565    | 0.04049   | 5.965     | 1.955e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:11] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 26        | 0.5565    | 0.009036  | 9.771     | 1.503e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:20] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 27        | 0.5517    | 0.02421   | 3.87      | 1.266e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:21] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 28        | 0.5439    | 0.01084   | 4.683     | 1.698e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:24] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 29        | 0.569     | 0.09279   | 8.834     | 381.6     |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:26] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 30        | 0.5614    | 0.1413    | 5.414     | 1.958e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:30] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 31        | 0.547     | 0.2411    | 3.293     | 117.6     |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:30] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 32        | 0.5739    | 0.1824    | 5.949     | 1.955e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:34] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 33        | 0.5405    | 0.01288   | 5.615     | 503.0     |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:36] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 34        | 0.5641    | 0.09231   | 6.351     | 786.8     |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:38] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 35        | 0.5391    | 0.2806    | 4.653     | 1.263e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:39] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 36        | 0.569     | 0.1475    | 9.847     | 1.057e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:43] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 37        | 0.5565    | 0.2468    | 7.127     | 418.9     |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:45] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 38        | 0.5641    | 0.1003    | 2.204     | 1.34e+03  |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:46] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 39        | 0.5614    | 0.005976  | 6.449     | 1.958e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:53] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 40        | 0.5424    | 0.03027   | 3.898     | 1.371e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:54] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 41        | 0.5439    | 0.2416    | 8.442     | 1.015e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:22:57] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 42        | 0.5614    | 0.2664    | 6.566     | 1.958e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:23:01] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 43        | 0.5487    | 0.1902    | 5.56      | 304.1     |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:23:02] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 44        | 0.5546    | 0.2444    | 6.036     | 946.8     |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:23:04] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 45        | 0.5739    | 0.2684    | 6.714     | 1.957e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:23:08] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 46        | 0.547     | 0.1472    | 6.284     | 1.957e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:23:12] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 47        | 0.5565    | 0.2648    | 5.978     | 1.41e+03  |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:23:14] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 48        | 0.5739    | 0.2066    | 6.654     | 1.957e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:23:18] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 49        | 0.5614    | 0.06074   | 9.435     | 1.057e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:23:22] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 50        | 0.5517    | 0.1788    | 8.994     | 381.6     |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:23:23] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 51        | 0.5641    | 0.1325    | 3.194     | 1.781e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:23:25] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 52        | 0.5913    | 0.2187    | 4.994     | 1.959e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:23:27] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 53        | 0.5739    | 0.0836    | 6.92      | 136.5     |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:23:28] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 54        | 0.547     | 0.2765    | 4.87      | 1.959e+03 |


/home/rik/.local/lib/python3.8/site-packages/xgboost/core.py:158: UserWarning: [13:23:31] WARNING: /workspace/src/learner.cc:740: 
Parameters: { "random_seed" } are not used.

  warnings.warn(smsg, UserWarning)


| 55        | 0.4727    | 0.01312   | 2.343     | 697.7     |


In [39]:
best_hyperparameters = bayesian_opt.max
best_hyperparameters['params'] = {param: int(value) if param in [
    'depth', 'iterations', 'n_estimators'] else value for param, value in best_hyperparameters['params'].items()}
print("Best hyperparameters:", best_hyperparameters['params'])
print(f"Best R-squared Score: {best_hyperparameters['target']:.4f}")

Best hyperparameters: {'learning_rate': 0.21871539472326787, 'max_depth': 4.99449273109376, 'n_estimators': 1958}
Best R-squared Score: 0.5913


In [40]:
params = {
    'max_depth': 4,            
    'learning_rate':0.21871539472326787
    ,  
    'n_estimators': 1958    
}
model_XGB = xg.XGBClassifier(**params, random_state=27)
model_XGB.fit(x_train_xgb, y_train_xgb)
y_pred_xgb = model_XGB.predict(x_train_xgb)
y_val_xgb = model_XGB.predict(x_valid_xgb)
print("Train f1_score: ", f1_score(y_train_xgb, y_pred_xgb))
print("Train accuracy_score: ", accuracy_score(y_train_xgb, y_pred_xgb))
print("Validation f1_score: ", f1_score(y_valid_xgb, y_val_xgb))
print("Validation accuracy_score: ", accuracy_score(y_valid_xgb, y_val_xgb))

Train f1_score:  0.9157088122605364
Train accuracy_score:  0.9432989690721649
Validation f1_score:  0.5913043478260869
Validation accuracy_score:  0.7589743589743589


После подбора гиперпараметров мы улучшили метрики и снизили переобучение для модели XGBClassifier на вадидации f1_score: 0.5913, accuracy_score: 0.759.

**Выводы**

Мы сравнили 4 регрессионые модели: DecisionTreeRegressor, CatBoostRegressor, LGBMRegressor, XGBRegressor. Наилучшие результаты показали модели LGBMClassifier с результатом на трейне: f1_score= 0.89 и accuracy_score= 0.93  и тесте: f1_score=0.5892, accuracy_score=0.764 и XGBClassifier с результатом на трейне: f1_score= 0.92 и accuracy_score= 0.94  и тесте: f1_score=0.5913, accuracy_score=0.759.

**Рекомендации**

Для улучшения значения метрики и снижения переобучения нужно будет увеличить количество данных.